<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Biomarker Discovery from High Throughput Screening Datasets • gxbioc2020workshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Biomarker Discovery from High Throughput Screening Datasets">
<meta property="og:description" content="gxbioc2020workshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gxbioc2020workshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.1</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../articles/GxWorkshop.html">Workshop</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/bhklab/bioc2020workshop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GxWorkshop_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Biomarker Discovery from High Throughput Screening Datasets</h1>
                        <h4 class="author">Petr Smirnov</h4>
            <address class="author_afil">
      Princess Margaret Cancer Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:psmirnov2000@gmail.com" class="email">psmirnov2000@gmail.com</a>
      </address>
                              <h4 class="author">Arvind Mer</h4>
            <address class="author_afil">
      Princess Margaret Cancer Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:arvind.mer@uhnresearch.ca" class="email">arvind.mer@uhnresearch.ca</a>
      </address>
                              <h4 class="author">Christopher Eeles</h4>
            <address class="author_afil">
      Princess Margaret Cancer Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:christopher.eeles@uhnresearch.ca" class="email">christopher.eeles@uhnresearch.ca</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/bhklab/bioc2020workshop/blob/master/vignettes/GxWorkshop.Rmd"><code>vignettes/GxWorkshop.Rmd</code></a></small>
      <div class="hidden name"><code>GxWorkshop.Rmd</code></div>

    </div>

    
    
<!-- Remove {-} from header to show numbering -->
<div id="workshop-overview" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#workshop-overview" class="anchor"></a>Workshop Overview</h1>
<div id="instructors-names-and-contact-information" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#instructors-names-and-contact-information" class="anchor"></a>Instructor(s) name(s) and contact information</h2>
<ul>
<li>Petr Smirnov <a href="mailto:petr.smirnov@uhnresearch.ca" class="email">petr.smirnov@uhnresearch.ca</a>
</li>
<li>Arvind Mer <a href="mailto:arvind.mer@uhnresearch.ca" class="email">arvind.mer@uhnresearch.ca</a>
</li>
<li>Christopher Eeles <a href="mailto:christopher.eeles@uhnresearch.ca" class="email">christopher.eeles@uhnresearch.ca</a>
</li>
</ul>
</div>
<div id="workshop-description" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>This workshop will focus on the synergies between analysis results from the <code>PharmacoGx</code>, <code>Xeva</code> and <code>RadioGx</code> packages and their usefulness for discovery of biomarkers of drug and/or radiation sensitivity in cancer cell lines (CCLs) and patent derived xenograft models (PDXs). We will discuss issues with data curation, consistency and reproducibility within the literature as well as illustrate the importance of unified analytical platforms, data and code sharing in bioinformatics and biomedical research. In this lab learners will be led through an analysis for each of the three packages on data provided by the download functions within them. The results of these analyses will then be explored to highlight how drug and radiation dose-response profiles in CCLs and PDXs can be used to discover potential synergistic biomarkers for drug-radiation and drug combination therapies. The resulting biomarkers will be discussed in the context of translational cancer research and clinical applications of genomic data. We will conclude with a discussion of how these biomarkers can be used to inform future <em>in vitro</em> and <em>in vivo</em> treatment screenings and ultimately provide useful insights for clinical trial design.</p>
<div id="pre-requisites" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<p>List any workshop prerequisites:</p>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Knowledge of or interest in pharmacogenomics and radio-genomics</li>
<li>Familiarity with base bioconductor objects, including the SummarizedExperiment Class</li>
<li>Familiarity with linear modelling in R; ANOVA, and goodness of fit tests</li>
</ul>
<p>List relevant background reading for the workshop, including any theoretical background you expect students to have.</p>
<ul>
<li>Smirnov, P. et al. PharmacoGx: An R package for analysis of large pharmacogenomic datasets. Bioinformatics 32, 1244–1246 (2016).</li>
<li>Manem, V. SK. et al. Modeling Cellular Response in Large-Scale Radiogenomic Databases to Advance Precision Radiotherapy. Cancer Research (2019) <a href="doi:10.1158/0008-5472.CAN-19-0179" class="uri">doi:10.1158/0008-5472.CAN-19-0179</a>.</li>
<li>Mer, A. S. et al. Integrative pharmacogenomics analysis of patient-derived xenografts. Cancer Res canres.0349.2019 (2019) <a href="doi:10.1158/0008-5472.CAN-19-0349" class="uri">doi:10.1158/0008-5472.CAN-19-0349</a>.</li>
</ul>
</div>
<div id="workshop-participation" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h3>
<p>Participants expected to have the following required packages installed on their machines to be able to run the commands along with the instructors. <code>PharmacoGx</code>, <code>Xeva</code>, <code>RadioGx</code>, <code>CoreGx</code> and <code>Biobase</code> from Bioconductor The workshop will be presented as a set of analysis steps to be replicated by the learners, with instructors available to explain the why and how of applying these functions to the given datasets. Learners will write analysis scripts as well as use interactive commands to explore the generated data structures and results. Learners will then brainstorm potential applications of the analysis results in their field as well as comment on use case examples presented by the instructors on research in our lab.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<p>Bioconductor:</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html" class="uri">https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/Xeva.html" class="uri">http://bioconductor.org/packages/release/bioc/html/Xeva.html</a></li>
<li>
<a href="https://cran.r-project.org/web/packages/RadioGx/index.html" class="uri">https://cran.r-project.org/web/packages/RadioGx/index.html</a>
<ul>
<li>RadioGx is currently available in the devel version of Bioconductor</li>
</ul>
</li>
<li>
<a href="https://cran.r-project.org/web/packages/CoreGx/index.html" class="uri">https://cran.r-project.org/web/packages/CoreGx/index.html</a>
<ul>
<li>CoreGx is currently available in the 3.11 release of Bioconductor</li>
</ul>
</li>
</ul>
</div>
</div>
<div id="time-outline" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h2>
<p>For a 2-hr workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>5m</td>
</tr>
<tr class="even">
<td>Data Structure and Common Functions Between Packages</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Basic functionalities of PharmacoGx</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Basic functionalities of Xeva</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Basic functionalities of RadioGx</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Data Analysis Walkthrough</td>
<td>30m</td>
</tr>
<tr class="odd">
<td>Understanding Analysis Results and Synergies</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Use Case Discussion: Translational Science</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h2>
<div id="learning-goals" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<ul>
<li>Describe pharmacogenomic datasets and radiogenomic datasets and usefulness in cancer research</li>
<li>Learn how to extract information from these datasets and to intersect them over their common measured features, experiments, model systems</li>
<li>Learn how to visualize experimental results from these datasets</li>
<li>Learn how to model dose-response for both small compound and radiotherapy datasets</li>
<li>Learn measures to quantify the response in both cell line and PDX sensitivity screens, for both drug and radiation treatments</li>
<li>Understand the differences in experimental design between radiotherapy and small compound treatments in cell lines</li>
<li>Understand the differences in measuring response to therapy between cell line and PDX based models</li>
<li>Understand how to choose the appropriate sensitivity quantification metric for your task</li>
<li>Understand how to identify potential genomic signatures of drug and radiation response and explore how synergies between these signatures can inform clinical trial design</li>
</ul>
</div>
<div id="learning-objectives" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>List available standardized pharmacogenomic and radiogenomic datasets and download them</li>
<li>Access the molecular features, dose-response and metadata contained within the data structures defined in the packages</li>
<li>Create drug-dose and radiation-dose response plots for cell line experiments</li>
<li>Create tumor growth curve plots for PDX experiments</li>
<li>Fit linear-quadratic models to radiotherapy experiments in cell lines</li>
<li>Fit Hill Slope models to dose-response experiments using small compound treatments in cell lines</li>
<li>Calculate the AAC, AUC, IC50, SF2, D10 metrics for response quantification in cell lines</li>
<li>Calculate the mRECIST, Slope, ABC, linear mix model and other tumour growth curve summary metrics</li>
<li>Visualize response of PDX experiments using an Oncoprint heatmap</li>
<li>Predict <em>in vivo</em> and <em>in vitro</em> univariate biomarkers using the PharmacoGx, Xeva and RadioGx packages</li>
<li>Compare signatures of response between chemotherapy compounds and radiotherapy</li>
<li>Compare univariate biomarkers between cell line and PDX experiments</li>
</ul>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This tutorial, titled Biomarker Discovery from High Throughput Screening Datasets, focuses on a suite of packages designed to simplify analysis correlating measures of treatment response measured on model systems with molecular profiles of those systems. These packages primarly focus on modeling response to cancer treatments in cancer models, but the concepts presented can be applied to other disease types, given availability of data for the area of interest.</p>
<p>The first two of the packages, <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em> and <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> focus on screens done using cell lines as a model system. PharmacoGx focuses on modeling and predicting the response of cell lines treated with small molecule therapies. RadioGx swaps the small molecule treatment for irradiation of cell lines. The third package, <em><a href="https://bioconductor.org/packages/3.12/Xeva">Xeva</a></em>, looks at treatment response for <strong>in vivo</strong> systems, primarily Patient Derived Xenographs of cancer tumours growing in immune compromised mice.</p>
<div id="data-structure-and-common-functions-between-packages" class="section level2">
<h2 class="hasAnchor">
<a href="#data-structure-and-common-functions-between-packages" class="anchor"></a>Data Structure and Common Functions Between Packages</h2>
<p>While the experiments analyzed by the three packages are different, the workflow that these packages accommodate has many similarities between them. First and foremost, all three packages were developed to provide data structures that package together screening and molecular data, as well as all related metadata, for complete studies. These packages also provide an interface to download large, publically available datasets which have standardized naming conventions, so that data across different studies is immediately comparable.</p>
<p>Furthermore, the analysis workflow shares common tasks across the three domains covered by these packages. This includes understanding and visualizing the data that is present in each dataset; modeling the response of the model systems to treatment; summarizing that response into a measure comparable across models; extracting and summarizing measurements of molecular features; and ultimately correlating the response of model systems to their molecular state.</p>
<p>In this tutorial, we will be highlighting each analysis step across all three packages, and then finish with a demonstration of how to integrate across the different domains presented here.</p>
</div>
</div>
<div id="experimental-designs" class="section level1">
<h1 class="hasAnchor">
<a href="#experimental-designs" class="anchor"></a>Experimental Designs</h1>
<p>RadioGx and PharmacoGx are both used to analyze data from <em>in vitro</em>, cell based assays. In cancer, the most common <em>in vitro</em> model used is the cancer cell line. These cell lines are usually obtained from patient tumour biopsies, which are passaged in a dish and often treated with factors to immortalize the cells - that is, to grant them infinite proliferation capacity. Such models are often deposited in repositories such as the ATCC, and made available for purchase to researchers who wish to use them in their experiments.</p>
<p><img src="CellLineFigure500ppi.png"></p>
<div id="in-vitro-pharmacogenomics" class="section level2">
<h2 class="hasAnchor">
<a href="#in-vitro-pharmacogenomics" class="anchor"></a><em>in vitro</em> Pharmacogenomics</h2>
<p>When studying Pharmacogenomics <em>in vitro</em>, the design of the experiment usually involves doing two things in parallel: Each cell line in the study is profiled on a molecular level, which can include measuring DNA level aberations such as SNVs and copy number changes, RNA expression levels, DNA methylation, and protein expression. In parallel, the cell lines undergo a viability based drug screen, where cell lines are treated with a panel of drugs, over a range of concentrations, and the growth of each cell line is compared to an untreated control.</p>
<p>The variable response to drug treatement across different cell lines is then compared with the molecular states of the cells, using machine learning or statistical approaches to discern which molecular features are predictive of the observed drug response.</p>
<p><img src="pharmacogx_design_exp.png"></p>
</div>
<div id="radiogenomics" class="section level2">
<h2 class="hasAnchor">
<a href="#radiogenomics" class="anchor"></a>Radiogenomics</h2>
<p>Radiogenomics follows a very similar design to Pharmacogenomics. The main difference is the form of the treatment applied to cells - in Pharmacogenomics, small molecule treatments are studied, while in radiogenomics, the treatement is irradiation of the cells. While the design of the study is similar, the difference in treatement causes differences in how the response of cells is modelled and quantified, as we detail below.</p>
<p>Currently, there is only one major, high-throughput dataset looking at <strong>in vitro</strong> radiogenomics, the Cleveland Clinic dataset. In this dataset, gamma radiation was used as the treatment. Conceptually however, there is no reason why alpha and beta radiation could not be studied in a similar experimental setup.</p>
<p><img src="radiogx_design_exp.png"></p>
</div>
<div id="in-vivo-pharmacogenomics" class="section level2">
<h2 class="hasAnchor">
<a href="#in-vivo-pharmacogenomics" class="anchor"></a><em>in vivo</em> Pharmacogenomics</h2>
<p><em>in vivo</em> models play an important role in cancer drug discovery. Patent derived xenografts (PDXs) are commonly used <em>in vivo</em> models in cancer research. They have several advantages over cell lines: * PDXs are known to recapitulate tumor biology very well * PDXs maintain the cellular and histological structure of the source tumor * PDXs also keep tumor micro-enviroment and stromal elements * Genomic properties of the tumor are largely preserved from patient to PDX * Drug response of PDXs closely match the corresponding patient and clinical outcome</p>
<p>To create PDXs, cancerous tissues or cells from patients’ tumors are engrafted into immunodeficient mice (either subcutaneous or orthotopic engraftment). Once established, tumors from the PDXs can be passed from mouse to mouse, this is called passaging. For drug testing, PDXs were given a set drug dose at a set schedule (for example weekly). Change in tumor size is measured, usually physically using calipers. <img src="PDXfigure500ppi.png"></p>
</div>
</div>
<div id="data-storage-and-access" class="section level1">
<h1 class="hasAnchor">
<a href="#data-storage-and-access" class="anchor"></a>Data Storage and Access</h1>
<div id="pharmacogx" class="section level2">
<h2 class="hasAnchor">
<a href="#pharmacogx" class="anchor"></a>PharmacoGx</h2>
<p>PharmacoGx stores drug screening data together with molecular profiling of cell lines in an object called a <code>PharmacoSet</code>, or <code>PSet</code> for short. This object inherits for the <code>CoreSet</code> class defined in our package <code>CoreGx</code>, which abstracts the common functionality across our similar packages. The basic structure of a PharmacoSet is as follows:</p>
<div class="figure">
<img src="PharmacoGxClassDiagram.png" alt="**PharmacoSet class diagram**. Objects comprising a `PharmacoSet` are enclosed in boxes. First box indicates type and name of each object. Second box indicates the structure of an object or class. Third box shows accessor methods from `PharmacoGx` for that specific object. '=&gt;' represents return and specifies what is returned from that item or method." width="1894"><p class="caption">
<strong>PharmacoSet class diagram</strong>. Objects comprising a <code>PharmacoSet</code> are enclosed in boxes. First box indicates type and name of each object. Second box indicates the structure of an object or class. Third box shows accessor methods from <code>PharmacoGx</code> for that specific object. ‘=&gt;’ represents return and specifies what is returned from that item or method.
</p>
</div>
<div id="downloading-pharmacoset-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#downloading-pharmacoset-objects" class="anchor"></a>Downloading PharmacoSet objects</h3>
<p>Let us download the PSets used for this tutorial into our local working environment. A table of available PharmacoSet objects can be obtained by using the <em>availablePSets</em> function.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r">  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">PharmacoGx</span>)
  <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/availablePSets.html">availablePSets</a></span>()</pre></body></html></div>
<p>If you look through the table produced by availablePSets, you will notice that each PSet object is downloaded from Zenodo - an Open Science project supported by the European Commision and CERN. Each PSet is assigned a DOI, which can be used to uniquely identify the data used in your analysis and retrieve it in the future for research reproducibility.</p>
<p>Any of the PharmacoSets in the table can then be downloaded by calling <code>downloadPSet</code>, which saves the datasets into a directory of the users choice, and returns the data into the R session. Lets try downloading the GDSC and CCLE PSets. The following code is included as an example, but not run, as the objects were preloaded onto the VM for the workshop.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r">  <span class="no">GDSC</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/downloadPSet.html">downloadPSet</a></span>(<span class="st">"GDSC_2020(v1-8.2)"</span>, <span class="kw">saveDir</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"."</span>, <span class="st">"PSets"</span>))
  <span class="no">CCLE</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/downloadPSet.html">downloadPSet</a></span>(<span class="st">"CCLE_2015"</span>, <span class="kw">saveDir</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"."</span>, <span class="st">"PSets"</span>))</pre></body></html></div>
<p>To load the datasets used in this workshop, we run the following code:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">GDSC</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">CCLE</span>)</pre></body></html></div>
<p>We can print out summary information about these datasets to the console, simply by evaluating them on the command line.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">CCLE</span></pre></body></html></div>
<pre><code>## Name:  CCLE 
## Date Created:  Mon Apr 27 00:20:19 2020 
## Number of cell lines:  1094 
## RNA: 
##  Dim:  10012 1036 
## Drug pertubation: 
##  Please look at pertNumber(cSet) to determine number of experiments for each drug-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  11670 
##  Please look at sensNumber(cSet) to determine number of experiments for each drug-cell combination.</code></pre>
</div>
</div>
<div id="radiogx" class="section level2">
<h2 class="hasAnchor">
<a href="#radiogx" class="anchor"></a>RadioGx</h2>
<p>The RadioSet has a structure similar to the <code>PharmacoSet</code> and also inherits from the CoreSet<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> class . The radiation slot is implemented in <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> to hold relevant metadata about the type(s) of radiation used in the dose-response experiment, and is analogous to the drug slot in a <code>PharmacoSet</code>. The remainder of the slots mirror the <code>PharmacoSet</code>.</p>
<div class="figure">
<img src="RadioGxClassDiagram.png" alt="**RadioSet class diagram**. Objects comprising a `RadioSet` are enclosed in boxes. First box indicates type and name of each object. Second box indicates the structure of an object or class. Third box shows accessor methods from `RadioGx` for that specific object. '=&gt;' represents return and specifies what is returned from that item or method." width="1894"><p class="caption">
<strong>RadioSet class diagram</strong>. Objects comprising a <code>RadioSet</code> are enclosed in boxes. First box indicates type and name of each object. Second box indicates the structure of an object or class. Third box shows accessor methods from <code>RadioGx</code> for that specific object. ‘=&gt;’ represents return and specifies what is returned from that item or method.
</p>
</div>
<div id="downloading-an-rset" class="section level3">
<h3 class="hasAnchor">
<a href="#downloading-an-rset" class="anchor"></a>Downloading an RSet</h3>
<p><em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> provides an interface similar to <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em> and <em><a href="https://bioconductor.org/packages/3.12/Xeva">Xeva</a></em> for downloading our curated versions of published datasets.</p>
<p>To get a list of available <code>RadioSet</code>s, use:</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="no">RSets</span> <span class="kw">&lt;-</span> <span class="fu">availableRSets</span>()</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">RadioSet_name</th>
<th align="left">Date_updated</th>
<th align="left">DOI</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">Cleveland</td>
<td align="left">2020/06/16</td>
<td align="left">10.5281/zenodo.3899555</td>
</tr></tbody>
</table>
<p>As the <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> package was only recently released, there is currently only one dataset available. Let’s download the ‘Cleveland’ RSet, which contains a highly curated version of the data from Yang <em>et al.</em>, 2016.</p>
<p>For this workshop, we have subset this dataset. The entire dataset can be downloaded using the following:</p>
<div class="sourceCode" id="cb7"><html><body><pre class="r"><span class="no">Cleveland</span> <span class="kw">&lt;-</span> <span class="fu">downloadRSet</span>(<span class="st">'Cleveland'</span>, <span class="kw">saveDir</span><span class="kw">=</span><span class="st">'data'</span>)</pre></body></html></div>
<p>Let’s load the RSet for this workshop.</p>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">Cleveland</span>)
<span class="no">Cleveland</span></pre></body></html></div>
<pre><code>## Name:  Cleveland 
## Date Created:  Wed Oct 25 17:38:42 2017 
## Number of cell lines:  303 
## Number of radiation types:  1 
## RNA: 
##  Dim:  20049 292 
## RNASeq: 
##  Dim:  61958 292 
## CNV: 
##  Dim:  24960 250 
## Drug pertubation: 
##  Please look at pertNumber(rSet) to determine number of
##             experiments for each radiation-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  305 
##  Please look at sensNumber(rSet) to determine number of
##             experiments for each radiation-cell combination.</code></pre>
<p>Similar to PharmacoGx and Xeva, a summary of the contents of the RadioSet is printed when calling a RadioSet in the console. We can see that the Cleveland RSet contains sensitivity information for 303 cell-lines treated with a single type of radiation. The RSet also contains rna<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, rna-seq and cnv molecular data for a subset of available cell-lines.</p>
</div>
</div>
<div id="xeva" class="section level2">
<h2 class="hasAnchor">
<a href="#xeva" class="anchor"></a>Xeva</h2>
<p>The R package <code>Xeva</code> connects the tumor volume vs. time data with genomics data in an R class called <code>XevaSet</code>. This allows users to analyze the data and perform biomarker discovery.</p>
<p>In a Xeva object, the <code>experiment</code> slot stores the data for each individual PDX/mouse. With the exception of tumor growth data (time vs. tumor volume), for each individual PDX/mouse, you can access metadata such as the patient’s age, sex, tissue histology, and passage information. All of this metadata is stored in the <code>pdxModel</code> class, where a unique ID called <code>model.id</code> is given to each PDX/mouse model. As for the tumor growth information, Xeva provides separate functions for retrieving and visualizing time vs. tumor volume data. We will see later how to get these data for an individual <code>model.id</code>, but first, let’s define some other terms that appear in the Xeva package.</p>
<p>A PDX experiment can be one of the two categories:</p>
<ul>
<li>
<em>treatment</em> represents experiments in which the PDX receives some kind of drug (or drug combination)</li>
<li>
<em>control</em> represents experiments in which the PDX receives no drug</li>
</ul>
<p>To see the effect of a drug, several replicate experiments are done for both the control and the treatment categories. In Xeva, a collection of PDX <code>model.ids</code> originating from the same patient is organized in batches (<code>batch</code>). A <code>batch</code> has two arms: <code>control</code> and <code>treatment</code>. This is illustrated in the figure below:</p>
<div class="figure">
<img src="Xeva_batch_2.png" alt="A PDX experiment. The text under each of the PDX/mouse (ie. m1, m2, p1, etc.) denotes the `model.id` in Xeva. In this example, three PDXs are delclared as control (m1, m2, and m3). Similarly, in the treatment arm, 3 PDXs are given the drug paclitaxel (p1, p2, and p3), 3 are given tamoxifen (t1, t2, and t3), and 3 are given binimetinib (b1, b2, b3). The PDXs in the control arm and one of the treatment arms together constitute a   extit{batch}. For example, control arm models (m1, m2, and m3) and treatment arm models (t1,t2, and t3) together create a batch called batch-2." width="1333"><p class="caption">
A PDX experiment. The text under each of the PDX/mouse (ie. m1, m2, p1, etc.) denotes the <code>model.id</code> in Xeva. In this example, three PDXs are delclared as control (m1, m2, and m3). Similarly, in the treatment arm, 3 PDXs are given the drug paclitaxel (p1, p2, and p3), 3 are given tamoxifen (t1, t2, and t3), and 3 are given binimetinib (b1, b2, b3). The PDXs in the control arm and one of the treatment arms together constitute a extit{batch}. For example, control arm models (m1, m2, and m3) and treatment arm models (t1,t2, and t3) together create a batch called batch-2.
</p>
</div>
<div id="downloading-xevaset-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#downloading-xevaset-objects" class="anchor"></a>Downloading Xevaset objects</h3>
<p>Several <code>XevaSet</code> objects can be obtained by using the <em>downloadXevaSet</em> function.</p>
<div class="sourceCode" id="cb10"><html><body><pre class="r">  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Xeva</span>)
  <span class="kw pkg">Xeva</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/downloadXevaSet.html">downloadXevaSet</a></span>()</pre></body></html></div>
<pre><code>##   XevaSet.Name
## 1    PDXE_BRCA
## 2      PDXE_CM
## 3     PDXE_CRC
## 4   PDXE_NSCLC
## 5    PDXE_PDAC
## 6         PDXE
##                                                                                                      Description
## 1                 XevaSet of breast cancer PDXs from the PDXE dataset. Source: Gao et al., Nature Medicine, 2015
## 2               XevaSet of cutaneous melanoma (CM) PDXs from the PDXE. Source: Gao et al., Nature Medicine, 2015
## 3               XevaSet of colorectal cancer (CRC) PDXs from the PDXE. Source: Gao et al., Nature Medicine, 2015
## 4 XevaSet of non-small-cell lung carcinoma (NSCLC) PDXs from the PDXE. Source: Gao et al., Nature Medicine, 2015
## 5    XevaSet of pancreatic ductal carcinoma (PDAC) PDXs from the PDXE. Source: Gao et al., Nature Medicine, 2015
## 6                   XevaSet of PDXE dataset containing all tissue type. Source: Gao et al., Nature Medicine 2015
##   Available.Molecular.Profiles             Date.Updated
## 1          RNASeq,mutation,cnv Fri Mar 15 16:58:41 2019
## 2          RNASeq,mutation,cnv Fri Mar 15 17:00:39 2019
## 3          RNASeq,mutation,cnv Fri Mar 15 17:01:15 2019
## 4          RNASeq,mutation,cnv Fri Mar 15 16:59:51 2019
## 5          RNASeq,mutation,cnv Fri Mar 15 17:02:20 2019
## 6          RNASeq,mutation,cnv Fri Mar 15 16:53:26 2019
##                                                                                           URL
## 1  https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/XevaSets/data/PDXE_BRCA.rds
## 2    https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/XevaSets/data/PDXE_CM.rds
## 3   https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/XevaSets/data/PDXE_CRC.rds
## 4 https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/XevaSets/data/PDXE_NSCLC.rds
## 5  https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/XevaSets/data/PDXE_PDAC.rds
## 6       https://www.pmgenomics.ca/bhklab/sites/default/files/downloads/XevaSets/data/PDXE.rds</code></pre>
<p>The following chunk loads the dataset used in the tutorial and prints out its summary:</p>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">brca_full</span>)
<span class="no">brca</span> <span class="kw">&lt;-</span> <span class="no">brca_full</span></pre></body></html></div>
</div>
</div>
</div>
<div id="examining-and-extracting-data-of-interest" class="section level1">
<h1 class="hasAnchor">
<a href="#examining-and-extracting-data-of-interest" class="anchor"></a>Examining and Extracting data of interest</h1>
<p>There are many common functions between the datasets for extracting the data relevant to your analysis. PharmacoGx are RadioGx are the most similar, due to the large similarities in their experimental designs. For most of your work with the objects, we recommend you use the accessor functions implemented in the packages to access the data. These accessors are largely stable across versions, while the implementation in the background can change as datasets grow in size and complexity.</p>
<p>This section is meant both as an introduction, and as a reference to the most used accessors for each package.</p>
<div id="coregx-accessors" class="section level2">
<h2 class="hasAnchor">
<a href="#coregx-accessors" class="anchor"></a>CoreGx accessors</h2>
<p>PharmacoGx and RadioGx share a lot of similarities in their object structure. This also means that a lot of the functions used to access the data provided in these objects can be accessed using the same interface.</p>
<p>In general, the PSet and RSet store three major categories of data: metadata/annotations, molecular data and radiation response data. These are demarcated in <strong>Fig.</strong> @ref(fig:pharmacoset) and <strong>Fig.</strong> @ref(fig:radioset) using green, blue and red, respectively. Accessor methods are available to retrieve all three kinds of data from a CoreSet; the accessor methods for each component are listed in the bottom most cell of each object in the class diagrams. We will discuss a subset of these methods now.</p>
<div id="accessing-metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-metadata" class="anchor"></a>Accessing metadata</h3>
<p>There are two major metadata slots in the <code>CoreSet</code> which are commonly useful for your analysis. The first is common between all the packages, the <code>cell</code> slot. This slot contains information about the cell lines which were profiled in the dataset. This information exists for all cell lines, if they exist in any of the molecular or sensitivity data included in the objects. Note that not all cell lines are profiled in all assays, which is one of the sources of complexity when analyzing data from large studies.</p>
<p>To access the cell line metadata, you can use the following accessors:</p>
<div class="sourceCode" id="cb13"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu">cellInfo</span>(<span class="no">CCLE</span>)[<span class="fl">1</span>:<span class="fl">3</span>,<span class="fl">1</span>:<span class="fl">3</span>])</pre></body></html></div>
<table class="table">
<colgroup>
<col width="5%">
<col width="5%">
<col width="63%">
<col width="24%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="left">cellid</th>
<th align="left">link</th>
<th align="left">CCLE.name</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">1321N1</td>
<td align="left">1321N1</td>
<td align="left"><a href="http://www.broadinstitute.org/ccle/cell%20lines/1321N1_CENTRAL_NERVOUS_SYSTEM" class="uri">http://www.broadinstitute.org/ccle/cell%20lines/1321N1_CENTRAL_NERVOUS_SYSTEM</a></td>
<td align="left">1321N1_CENTRAL_NERVOUS_SYSTEM</td>
</tr>
<tr class="even">
<td align="left">143B</td>
<td align="left">143B</td>
<td align="left"><a href="http://www.broadinstitute.org/ccle/cell%20lines/143B_BONE" class="uri">http://www.broadinstitute.org/ccle/cell%20lines/143B_BONE</a></td>
<td align="left">143B_BONE</td>
</tr>
<tr class="odd">
<td align="left">22Rv1</td>
<td align="left">22Rv1</td>
<td align="left"><a href="http://www.broadinstitute.org/ccle/cell%20lines/22RV1_PROSTATE" class="uri">http://www.broadinstitute.org/ccle/cell%20lines/22RV1_PROSTATE</a></td>
<td align="left">22RV1_PROSTATE</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu">cellInfo</span>(<span class="no">Cleveland</span>)[<span class="fl">1</span>:<span class="fl">3</span>,<span class="fl">1</span>:<span class="fl">3</span>])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">cellid</th>
<th align="left">tissueid</th>
<th align="left">CellLine</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">SK-N-FI</td>
<td align="left">SK-N-FI</td>
<td align="left">autonomic_ganglia</td>
<td align="left">SKNFI</td>
</tr>
<tr class="even">
<td align="left">SK-N-AS</td>
<td align="left">SK-N-AS</td>
<td align="left">autonomic_ganglia</td>
<td align="left">SKNAS</td>
</tr>
<tr class="odd">
<td align="left">CHP-212</td>
<td align="left">CHP-212</td>
<td align="left">autonomic_ganglia</td>
<td align="left">CHP212</td>
</tr>
</tbody>
</table>
<p>The standardized identifiers for those cell lines across our packages are the row names of this object. For convenience, they are accessible through the <code>cellNames</code> function:</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">cellNames</span>(<span class="no">CCLE</span>))</pre></body></html></div>
<pre><code>## [1] "1321N1"   "143B"     "22Rv1"    "23132/87" "253J"     "253J-BV"</code></pre>
<p>For example, here we can use the <code>cellInfo</code> function to pull out the tissue of origin for each cell line in the CCLE dataset. The <code>tissueid</code> column is standard for PSets, and should be present in all objects.</p>
<div class="sourceCode" id="cb17"><html><body><pre class="r"><span class="no">mycol</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#8dd3c7"</span>,<span class="st">"#ffffb3"</span>,<span class="st">"#bebada"</span>,<span class="st">"#fb8072"</span>,<span class="st">"#80b1d3"</span>,<span class="st">"#fdb462"</span>,
           <span class="st">"#b3de69"</span>,<span class="st">"#fccde5"</span>,<span class="st">"#d9d9d9"</span>,<span class="st">"#bc80bd"</span>,<span class="st">"#ccebc5"</span>,<span class="st">"#ffed6f"</span>,
           <span class="st">"#a6cee3"</span>,<span class="st">"#1f78b4"</span>,<span class="st">"#b2df8a"</span>,<span class="st">"#33a02c"</span>,<span class="st">"#fb9a99"</span>,<span class="st">"#e31a1c"</span>,
           <span class="st">"#fdbf6f"</span>,<span class="st">"#ff7f00"</span>,<span class="st">"#cab2d6"</span>,<span class="st">"#6a3d9a"</span>,<span class="st">"#ffff99"</span>,<span class="st">"#b15928"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/pie.html">pie</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">CCLE</span>@<span class="kw">cell</span>[,<span class="st">"tissueid"</span>]),
    <span class="kw">col</span><span class="kw">=</span><span class="no">mycol</span>,
    <span class="kw">main</span><span class="kw">=</span><span class="st">"Tissue types"</span>,
    <span class="kw">radius</span><span class="kw">=</span><span class="fl">1</span>,
    <span class="kw">cex</span><span class="kw">=</span><span class="fl">0.8</span>)</pre></body></html></div>
<div class="figure">
<img src="GxWorkshop_files/figure-html/piechart-1.png" alt="Tissue of origin of cell lines in CCLE study" width="700"><p class="caption">
Tissue of origin of cell lines in CCLE study
</p>
</div>
<p>The other important metadata slot will vary across the two packages, being called <code>drug</code> in PharmacoGx, and <code>radiation</code> in RadioGx. They are discussed just below.</p>
<p>The <code>CoreSet</code> also includes <code>annotations</code>, which keep track of metadata about the creation of the <code>CoreSet</code> object itself, and <code>curation</code>, which keeps track of the remapping of identifiers from those used in the original study to those in the object.</p>
</div>
<div id="accessing-molecular-data" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-molecular-data" class="anchor"></a>Accessing molecular data</h3>
<p>Molecular data are contained in the <code>molecularProfiles</code> slot.</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="co"># Get the list (equivalent to @molecularProfiles, except that it is robust to changes in CoreSet structure</span>
<span class="co">## PSet</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="fu">molecularProfilesSlot</span>(<span class="no">CCLE</span>), <span class="kw">max.level</span><span class="kw">=</span><span class="fl">2</span>)</pre></body></html></div>
<pre><code>## List of 1
##  $ rna:Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots</code></pre>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="co">## RSet</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="fu">molecularProfilesSlot</span>(<span class="no">Cleveland</span>), <span class="kw">max.level</span><span class="kw">=</span><span class="fl">2</span>)</pre></body></html></div>
<pre><code>## List of 4
##  $ rna     :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##  $ rnaseq  :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##  $ mutation:Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##  $ cnv     :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots</code></pre>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="co"># Get the names from the list</span>
<span class="fu">mDataNames</span>(<span class="no">CCLE</span>)</pre></body></html></div>
<pre><code>## [1] "rna"</code></pre>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="fu">mDataNames</span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<pre><code>## [1] "rna"      "rnaseq"   "mutation" "cnv"</code></pre>
<p>All molecular data in any class inheriting from CoreSet is contained in a <code>SummarizedExperiment</code> object. While <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> comes with it’s own set of accessors, we recommend using available <code>RadioGx</code> and <code>PharmacoGx</code> accessor methods as it allows your scripts to be robust to future changes in the structure of a <code>CoreSet</code> object. Below, we demonstrate each method with an RSet, but encourage you to try them on the PSets as well. They should all work!</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="co"># Get sample metadata</span>
<span class="no">phenoInf</span> <span class="kw">&lt;-</span> <span class="fu">phenoInfo</span>(<span class="no">Cleveland</span>, <span class="st">'cnv'</span>)</pre></body></html></div>
<p>To keep the document formatted nicely, the following tables have been subset to the first three rows and columns.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Sample_title</th>
<th align="left">Sample_geo_accession</th>
<th align="left">Sample_status</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">A2780</td>
<td align="left">GSM887918</td>
<td align="left">Public on Mar 20 2012</td>
</tr>
<tr class="even">
<td align="left">OVCAR-4</td>
<td align="left">GSM888565</td>
<td align="left">Public on Mar 20 2012</td>
</tr>
<tr class="odd">
<td align="left">VMRC-RCZ</td>
<td align="left">GSM888829</td>
<td align="left">Public on Mar 20 2012</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb27"><html><body><pre class="r"><span class="co"># Get feature metadata</span>
<span class="no">featInfo</span> <span class="kw">&lt;-</span> <span class="fu">featureInfo</span>(<span class="no">Cleveland</span>, <span class="st">'rna'</span>)</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">Probe</th>
<th align="left">EnsemblGeneId</th>
<th align="right">EntrezGeneId</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000000003_at</td>
<td align="left">ENSG00000000003</td>
<td align="right">7105</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000005_at</td>
<td align="left">ENSG00000000005</td>
<td align="right">64102</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000419_at</td>
<td align="left">ENSG00000000419</td>
<td align="right">8813</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="co"># Access the moleclar feature data</span>
<span class="no">mProf</span> <span class="kw">&lt;-</span> <span class="fu">molecularProfiles</span>(<span class="no">Cleveland</span>, <span class="st">'rnaseq'</span>)</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">G20461.HSC-3.2</th>
<th align="right">G20463.C2BBe1.2</th>
<th align="right">G20466.5637.2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000000003</td>
<td align="right">4.527</td>
<td align="right">5.860</td>
<td align="right">5.394</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000005</td>
<td align="right">0.000</td>
<td align="right">0.070</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000419</td>
<td align="right">6.314</td>
<td align="right">7.303</td>
<td align="right">6.140</td>
</tr>
</tbody>
</table>
</div>
<div id="accessing-response-data" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-response-data" class="anchor"></a>Accessing response data</h3>
<p>Data for treatment response can also be accessed the same way across <code>PharmacoGx</code> and <code>RadioGx</code>. The difference between the two packages is in the data itself, and the models applied to the data, discussed below. Here is a quick summary of accessor functions for response data.</p>
<div class="sourceCode" id="cb29"><html><body><pre class="r"><span class="co"># Get sensitivity slot</span>
<span class="no">sens</span> <span class="kw">&lt;-</span> <span class="fu">sensitivitySlot</span>(<span class="no">CCLE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb30"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">sens</span>)</pre></body></html></div>
<pre><code>## [1] "info"     "raw"      "profiles" "n"</code></pre>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="co"># Get sensitivity raw data</span>
<span class="no">sensRaw</span> <span class="kw">&lt;-</span> <span class="fu">sensitivityRaw</span>(<span class="no">CCLE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb33"><html><body><pre class="r"><span class="co"># Get sensitivity raw data</span>
<span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">sensRaw</span>[,,<span class="fl">2</span>][<span class="fl">1</span>:<span class="fl">3</span>, <span class="fl">1</span>:<span class="fl">3</span>])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">doses1</th>
<th align="right">doses2</th>
<th align="right">doses3</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">drugid_AEW541_1321N1</td>
<td align="right">108.67</td>
<td align="right">111.00</td>
<td align="right">102.16</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_22Rv1</td>
<td align="right">100.94</td>
<td align="right">112.50</td>
<td align="right">86.00</td>
</tr>
<tr class="odd">
<td align="left">drugid_AEW541_42-MG-BA</td>
<td align="right">108.91</td>
<td align="right">108.39</td>
<td align="right">96.50</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="co"># Get sensitivity profiles</span>
<span class="no">sensProf</span> <span class="kw">&lt;-</span> <span class="fu">sensitivityProfiles</span>(<span class="no">CCLE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb35"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">sensProf</span>[<span class="fl">1</span>:<span class="fl">3</span>, <span class="fl">1</span>:<span class="fl">3</span>])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">ic50_published</th>
<th align="right">aac_published</th>
<th align="right">amax_published</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">drugid_AEW541_1321N1</td>
<td align="right">8.000000</td>
<td align="right">0.0873750</td>
<td align="right">-42.55801</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_22Rv1</td>
<td align="right">2.329924</td>
<td align="right">0.2205000</td>
<td align="right">-71.58934</td>
</tr>
<tr class="odd">
<td align="left">drugid_AEW541_42-MG-BA</td>
<td align="right">2.682130</td>
<td align="right">0.1144375</td>
<td align="right">-63.49137</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="co"># Get sensitivity info</span>
<span class="no">sensInfo</span> <span class="kw">&lt;-</span> <span class="fu">sensitivityInfo</span>(<span class="no">CCLE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">sensInfo</span>[<span class="fl">1</span>:<span class="fl">3</span>, <span class="fl">1</span>:<span class="fl">3</span>])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">cellid</th>
<th align="left">drugid</th>
<th align="right">nbr.conc.tested</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">drugid_AEW541_1321N1</td>
<td align="left">1321N1</td>
<td align="left">Nvp-aew541</td>
<td align="right">8</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_22Rv1</td>
<td align="left">22Rv1</td>
<td align="left">Nvp-aew541</td>
<td align="right">8</td>
</tr>
<tr class="odd">
<td align="left">drugid_AEW541_42-MG-BA</td>
<td align="left">42-MG-BA</td>
<td align="left">Nvp-aew541</td>
<td align="right">8</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="pharmacogx-drug-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#pharmacogx-drug-metadata" class="anchor"></a>PharmacoGx Drug Metadata</h2>
<p>In PharmacoGx, metadata about the compounds used can be accessed through the <code>drugInfo</code> accessor. This works very similar to the <code>cellInfo</code> function.</p>
<div class="sourceCode" id="cb38"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugInfo.html">drugInfo</a></span>(<span class="no">CCLE</span>)[<span class="fl">1</span>:<span class="fl">3</span>,<span class="fl">1</span>:<span class="fl">3</span>])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">Compound..code.or.generic.name.</th>
<th align="left">Compound..brand.name.</th>
<th align="left">Target.s.</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Crizotinib</td>
<td align="left">PF-2341066</td>
<td align="left">Crizotinib</td>
<td align="left">c-MET, ALK</td>
</tr>
<tr class="even">
<td align="left">Dovitinib</td>
<td align="left">TKI258</td>
<td align="left">Dovitinib</td>
<td align="left">EGFR, FGFR1, PDGFRbeta, VEGFR-1, KDR</td>
</tr>
<tr class="odd">
<td align="left">Erlotinib</td>
<td align="left">Erlotinib</td>
<td align="left">Tarceva</td>
<td align="left">EGFR</td>
</tr>
</tbody>
</table>
<p>Drug names are also accessed as one would expect:</p>
<div class="sourceCode" id="cb39"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugNames.html">drugNames</a></span>(<span class="no">CCLE</span>)</pre></body></html></div>
<pre><code>##  [1] "Crizotinib"       "Dovitinib"        "Erlotinib"        "Irinotecan"      
##  [5] "L-685458"         "Lapatinib"        "LBW242"           "Nilotinib"       
##  [9] "Nutlin-3"         "Nvp-aew541"       "Nvp-tae 684"      "Paclitaxel"      
## [13] "Palbociclib"      "Panobinostat"     "PD-0325901"       "Pha-665752"      
## [17] "PLX4720"          "Raf265(chir-265)" "Saracatinib"      "Selumetinib"     
## [21] "Sorafenib"        "Tanespimycin"     "Topotecan"        "Vandetanib"</code></pre>
</div>
<div id="radiation-metadata" class="section level2">
<h2 class="hasAnchor">
<a href="#radiation-metadata" class="anchor"></a>Radiation Metadata</h2>
<p>A unique slot to RadioGx, <code>radiation</code> has additional accessor methods to retrieve the radiation types used in a given sensitivity experiment.</p>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="co"># Get the radiation info data.frame from an RSet</span>
<span class="no">radInf</span> <span class="kw">&lt;-</span> <span class="fu">radiationInfo</span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<div class="sourceCode" id="cb42"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">radInf</span>)</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="left">X.radiation.</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">radiation</td>
<td align="left">radiation</td>
</tr></tbody>
</table>
<p>Currently, only one type of radiation has been used in an <code>RSet</code>. However, we hope to add new <code>RSet</code>s covering a wider range of radiation sensitivity and perturbation experiments in the near future. The following method is also available to retrieve the radiation types as a <code>character</code> vector instead of a <code>data.frame</code>.</p>
<div class="sourceCode" id="cb43"><html><body><pre class="r"><span class="no">radTypes</span> <span class="kw">&lt;-</span> <span class="fu">radiationTypes</span>(<span class="no">Cleveland</span>)
<span class="no">radTypes</span></pre></body></html></div>
<pre><code>## [1] "radiation"</code></pre>
</div>
<div id="subsetting-pharmacosets" class="section level2">
<h2 class="hasAnchor">
<a href="#subsetting-pharmacosets" class="anchor"></a>Subsetting PharmacoSets</h2>
<p>PSets can be subsetted by refering directly to the drugs and cell lines you want to keep in the dataset. For example, we can subset the GDSC PSet in the following ways:</p>
<div class="sourceCode" id="cb45"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">GDSC</span>)</pre></body></html></div>
<pre><code>## Name:  GDSC_v1 
## Date Created:  Fri Jun  5 20:29:32 2020 
## Number of cell lines:  1064 
## RNA: 
##  Dim:  5947 789 
## Drug pertubation: 
##  Please look at pertNumber(cSet) to determine number of experiments for each drug-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  6890 
##  Please look at sensNumber(cSet) to determine number of experiments for each drug-cell combination.</code></pre>
<div class="sourceCode" id="cb47"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugNames.html">drugNames</a></span>(<span class="no">GDSC</span>)[<span class="fl">1</span>])</pre></body></html></div>
<pre><code>## [1] "Paclitaxel"</code></pre>
<div class="sourceCode" id="cb49"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu">subsetTo</span>(<span class="no">GDSC</span>, <span class="kw">drugs</span><span class="kw">=</span><span class="st">"Erlotinib"</span>))</pre></body></html></div>
<pre><code>## Name:  GDSC_v1 
## Date Created:  Fri Jun  5 20:29:32 2020 
## Number of cell lines:  824 
## RNA: 
##  Dim:  5947 789 
## Drug pertubation: 
##  Please look at pertNumber(cSet) to determine number of experiments for each drug-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  511 
##  Please look at sensNumber(cSet) to determine number of experiments for each drug-cell combination.</code></pre>
<div class="sourceCode" id="cb51"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">GDSC</span>[<span class="st">"YT"</span>,<span class="st">"Erlotinib"</span>])</pre></body></html></div>
<pre><code>## Name:  GDSC_v1 
## Date Created:  Fri Jun  5 20:29:32 2020 
## Number of cell lines:  1 
## RNA: 
##  Dim:  5947 0 
## Drug pertubation: 
##  Please look at pertNumber(cSet) to determine number of experiments for each drug-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  1 
##  Please look at sensNumber(cSet) to determine number of experiments for each drug-cell combination.</code></pre>
<p>As an exercise, lets manually compute the number of overlaping cell lines between CCLE and GDSC. Try it yourself!</p>
<p>If you noticed as you completed the previous exercise, cell line and drug names are standardized between PSets downloaded using the <em>downloadPSet</em> function. This is the major benefit in working with the pre-curated objects: entities follow a standardized naming scheme, which was chosen to be human readable whenever possible.</p>
<p>PharmacoGx also implements a convenience function to intersect PSets:</p>
<div class="sourceCode" id="cb53"><html><body><pre class="r"><span class="no">common</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/intersectPSet.html">intersectPSet</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">intersectOn</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"cell.lines"</span>))
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">common</span>)</pre></body></html></div>
</div>
<div id="xeva-1" class="section level2">
<h2 class="hasAnchor">
<a href="#xeva-1" class="anchor"></a>Xeva</h2>
<p>The structure of Xeva is different from the CoreGx structure. As such, many of the accessor functions are also different.</p>
<p>As mentioned earlier, Xeva stores metadata for each individual PDX model. We can retrieve the meta-information about each PDX, such as number of models and tissue type, using:</p>
<div class="sourceCode" id="cb54"><html><body><pre class="r"><span class="no">brca.mod</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/modelInfo.html">modelInfo</a></span>(<span class="no">brca</span>)
<span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">brca.mod</span>)</pre></body></html></div>
<pre><code>## [1] 849   5</code></pre>
<div class="sourceCode" id="cb56"><html><body><pre class="r"><span class="no">brca.mod</span>[<span class="fl">1</span>:<span class="fl">4</span>, ]</pre></body></html></div>
<pre><code>##                model.id tissue   tissue.name patient.id        drug
## X.1004.BG98 X.1004.BG98   BRCA Breast Cancer     X-1004      BGJ398
## X.1004.biib X.1004.biib   BRCA Breast Cancer     X-1004 binimetinib
## X.1004.BK20 X.1004.BK20   BRCA Breast Cancer     X-1004      BKM120
## X.1004.BY19 X.1004.BY19   BRCA Breast Cancer     X-1004      BYL719</code></pre>
<p>The output shows that the <code>brca</code> dataset contains 849 PDX models. We can also see the time vs. tumor volume data for a model using:</p>
<div class="sourceCode" id="cb58"><html><body><pre class="r"><span class="no">model.data</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/getExperiment.html">getExperiment</a></span>(<span class="no">brca</span>, <span class="kw">model.id</span> <span class="kw">=</span> <span class="st">"X.1004.BG98"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">model.data</span>)</pre></body></html></div>
<pre><code>##      model.id drug.join.name time volume body.weight volume.normal
## 1 X.1004.BG98         BGJ398    0  199.7        28.2     0.0000000
## 2 X.1004.BG98         BGJ398    2  181.9        28.0    -0.0891337
## 3 X.1004.BG98         BGJ398    5  172.7        28.4    -0.1352028
## 4 X.1004.BG98         BGJ398    9  129.6        27.2    -0.3510265
## 5 X.1004.BG98         BGJ398   12   91.3        26.7    -0.5428142
## 6 X.1004.BG98         BGJ398   16  117.1        26.2    -0.4136204</code></pre>
<p>Similarly, for <code>batch</code> names, we can obtain all predefined batch names using:</p>
<div class="sourceCode" id="cb60"><html><body><pre class="r"><span class="no">batch.name</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/batchInfo.html">batchInfo</a></span>(<span class="no">brca</span>)
<span class="no">batch.name</span>[<span class="fl">1</span>:<span class="fl">4</span>]</pre></body></html></div>
<pre><code>## [1] "X-1004.BGJ398"      "X-1004.binimetinib" "X-1004.BKM120"     
## [4] "X-1004.BYL719"</code></pre>
<p>The information about a  can be shown using:</p>
<div class="sourceCode" id="cb62"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/batchInfo.html">batchInfo</a></span>(<span class="no">brca</span>, <span class="kw">batch</span> <span class="kw">=</span> <span class="st">"X-1004.binimetinib"</span>)</pre></body></html></div>
<pre><code>## $`X-1004.binimetinib`
## name = X-1004.binimetinib
## control = X.1004.uned
## treatment = X.1004.biib</code></pre>
<p>Here, for the batch named <code>X-1004.binimetinib</code>, we can see that the control sample is <code>X.1004.uned</code> and the treatment sample is <code>X.1004.biib</code>.</p>
</div>
</div>
<div id="visualizing-and-modeling-sensitivity-data" class="section level1">
<h1 class="hasAnchor">
<a href="#visualizing-and-modeling-sensitivity-data" class="anchor"></a>Visualizing and modeling sensitivity data</h1>
<div id="working-with-dose-response-data-from-psets" class="section level2">
<h2 class="hasAnchor">
<a href="#working-with-dose-response-data-from-psets" class="anchor"></a>Working with Dose Response data from PSets</h2>
<div id="plotting-drug-dose-response-data" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-drug-dose-response-data" class="anchor"></a>Plotting Drug Dose Response Data</h3>
<p>Drug-Dose response data included in the PharmacoSet objects can be conviniently plotted using the <code>drugDoseResponseCurve</code> function. Given a list of PharmacoSets, a drug name and a cell name, it will plot the drug dose response curves for the given cell-drug combination in each dataset, allowing direct comparisons of data between datasets.</p>
<div class="sourceCode" id="cb64"><html><body><pre class="r"><span class="no">cells</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"SK-MEL-2"</span>,<span class="st">"697"</span>,<span class="st">"NCI-H1666"</span>)
<span class="co"># par(mfrow=c(1, 3), pty = "s")</span>
<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">drug</span><span class="kw">=</span><span class="st">"Lapatinib"</span>, <span class="kw">cellline</span><span class="kw">=</span><span class="no">cells</span>[<span class="fl">1</span>],
                      <span class="kw">pSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Actual"</span>,
                      <span class="kw">legends.label</span><span class="kw">=</span><span class="st">"aac_published"</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/curves-1.png" width="700" style="display: block; margin: auto auto auto 0;"></p>
<div class="sourceCode" id="cb65"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">drug</span><span class="kw">=</span><span class="st">"Lapatinib"</span>, <span class="kw">cellline</span><span class="kw">=</span><span class="no">cells</span>[<span class="fl">2</span>],
                      <span class="kw">pSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Actual"</span>,
                      <span class="kw">legends.label</span><span class="kw">=</span><span class="st">"aac_published"</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/curves-2.png" width="700" style="display: block; margin: auto auto auto 0;"></p>
<div class="sourceCode" id="cb66"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">drug</span><span class="kw">=</span><span class="st">"Lapatinib"</span>, <span class="kw">cellline</span><span class="kw">=</span><span class="no">cells</span>[<span class="fl">3</span>],
                      <span class="kw">pSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Actual"</span>,
                      <span class="kw">legends.label</span><span class="kw">=</span><span class="st">"aac_published"</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/curves-3.png" width="700" style="display: block; margin: auto auto auto 0;"></p>
<p>The function <code>drugDoseResponseCurve</code> can also be used to plot your own drug dose response curves, as follows:</p>
<div class="sourceCode" id="cb67"><html><body><pre class="r"><span class="no">concentrations</span> <span class="kw">&lt;-</span> <span class="fl">1</span>/<span class="fl">2</span>^<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">8</span>) * <span class="fl">1</span>
<span class="no">viabilities</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">33.3</span>, <span class="fl">60</span>, <span class="fl">77.8</span>, <span class="fl">88.2</span>, <span class="fl">93.9</span>, <span class="fl">96.9</span>, <span class="fl">98.4</span>, <span class="fl">99.2</span>)

<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">concentrations</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Exp 1"</span> <span class="kw">=</span> <span class="no">concentrations</span>), <span class="kw">viabilities</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Exp 1"</span> <span class="kw">=</span> <span class="no">viabilities</span>))</pre></body></html></div>
<pre><code>## Warning in sanitizeInput(concentrations[[i]], viabilities[[i]], conc_as_log =
## conc_as_log, : Concentration Values were unsorted. Sorting concentration and
## ordering viability in same order</code></pre>
<p><img src="GxWorkshop_files/figure-html/ddrc_own-1.png" width="700"></p>
</div>
<div id="fitting-drug-dose-response-curves" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-drug-dose-response-curves" class="anchor"></a>Fitting Drug Dose Response Curves</h3>
<p>One of the core tasks implemented in PharmacoGx is the fitting of Hill Curve Models to dose-response data. In PharmacoGx, we use the 3 Parameter Hill Curve function as our model of drug response in cancer cell lines:</p>
<p><span class="math display">\[y = E_\infty + \frac{1-E_\infty}{1 + (\frac{x}{EC50})^{HS}}\]</span> This is a log-logistic model, meaning that it takes the form of a logistic curve when x is on the log scale. The three parameters are: <span class="math inline">\(E_\infty\)</span>, which is the maximal inhibition predicted at infinite concetration of the drug (the right asymptote); <span class="math inline">\(EC50\)</span>, which is the inflection point of the logit, where the inhibition reaches 50% of the maximum; <span class="math inline">\(HS\)</span>, the Hill Slope, which is a measure of the cooperativity of binding. This parameter controls the steepness of the logit, and is interpreted based on its relation to 1. Values larger than 1 imply positive cooperativity in inhibition/binding of the target for the small molcule ligands, which values less than 1 imply negative cooperativity/antagonism.</p>
<p>To fit a Hill Slope model to your data in PharmacoGx, you can use the <code>logLogisticRegression</code> function, as below:</p>
<div class="sourceCode" id="cb69"><html><body><pre class="r"><span class="no">concentrations</span> <span class="kw">&lt;-</span> <span class="fl">1</span>/<span class="fl">2</span>^<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">8</span>) * <span class="fl">1</span>
<span class="no">viabilities</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">33.3</span>, <span class="fl">60</span>, <span class="fl">77.8</span>, <span class="fl">88.2</span>, <span class="fl">93.9</span>, <span class="fl">96.9</span>, <span class="fl">98.4</span>, <span class="fl">99.2</span>)

<span class="no">pars</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/logLogisticRegression.html">logLogisticRegression</a></span>(<span class="kw">conc</span> <span class="kw">=</span> <span class="no">concentrations</span>, <span class="kw">viability</span> <span class="kw">=</span> <span class="no">viabilities</span>)</pre></body></html></div>
<pre><code>## Warning in sanitizeInput(conc = conc, viability = viability, conc_as_log =
## conc_as_log, : Concentration Values were unsorted. Sorting concentration and
## ordering viability in same order</code></pre>
<div class="sourceCode" id="cb71"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">pars</span>)</pre></body></html></div>
<pre><code>## $HS
## [1] 1.900066
## 
## $E_inf
## [1] 0
## 
## $EC50
## [1] 0.2813364</code></pre>
<p>For all the data in PSets, this modeling has been precomputed and stored in the object, under <code>sensitivityProfiles</code>:</p>
<div class="sourceCode" id="cb73"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu">sensitivityProfiles</span>(<span class="no">CCLE</span>)[<span class="fl">1</span>:<span class="fl">5</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"E_inf"</span>, <span class="st">"EC50"</span>, <span class="st">"HS"</span>)])</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">E_inf</th>
<th align="right">EC50</th>
<th align="right">HS</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">drugid_AEW541_1321N1</td>
<td align="right">38.47711</td>
<td align="right">3.295213</td>
<td align="right">0.9169196</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_22Rv1</td>
<td align="right">13.95620</td>
<td align="right">1.376470</td>
<td align="right">0.8374151</td>
</tr>
<tr class="odd">
<td align="left">drugid_AEW541_42-MG-BA</td>
<td align="right">36.28378</td>
<td align="right">1.723197</td>
<td align="right">2.9099240</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_5637</td>
<td align="right">0.00000</td>
<td align="right">5.192732</td>
<td align="right">1.0883419</td>
</tr>
<tr class="odd">
<td align="left">drugid_AEW541_639-V</td>
<td align="right">30.29307</td>
<td align="right">1.350122</td>
<td align="right">0.6991911</td>
</tr>
</tbody>
</table>
</div>
<div id="computing-summary-measures-for-ddrcs" class="section level3">
<h3 class="hasAnchor">
<a href="#computing-summary-measures-for-ddrcs" class="anchor"></a>Computing Summary Measures for DDRCs</h3>
<p>Often with <em>in vitro</em> Pharmacogenomics data, we want to compare the drug sensitivity of a cell line to some omic feature. For this, we want to summarize the drug dose response curve into a single number representing the sensitivity of the cell line. The IC50 and Area Above the Curve are two convenient metrics for quantifying the observed drug sensitivity. If you noticed above, the <code>drugDoseResponseCurve</code> function computes them by default. In PharmacoGx, they can be computed manually as follows:</p>
<div class="sourceCode" id="cb74"><html><body><pre class="r"><span class="no">concentrations</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="fl">1</span>/<span class="fl">2</span>^<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">8</span>) * <span class="fl">1</span>)
<span class="no">viabilities</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">99.2</span>, <span class="fl">98.4</span>, <span class="fl">96.9</span>, <span class="fl">93.9</span>, <span class="fl">88.2</span>, <span class="fl">77.8</span>, <span class="fl">60</span>, <span class="fl">33.3</span>, <span class="fl">0</span>)

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeAUC.html">computeAUC</a></span>(<span class="kw">concentration</span> <span class="kw">=</span> <span class="no">concentrations</span>, <span class="kw">viability</span> <span class="kw">=</span> <span class="no">viabilities</span>))</pre></body></html></div>
<pre><code>## [1] 23.68416</code></pre>
<div class="sourceCode" id="cb76"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeICn.html">computeIC50</a></span>(<span class="kw">concentration</span> <span class="kw">=</span> <span class="no">concentrations</span>, <span class="kw">viability</span> <span class="kw">=</span> <span class="no">viabilities</span>))</pre></body></html></div>
<pre><code>## [1] 0.2813364</code></pre>
<p>In PharmacoGx, we call these measures sensitivity measures. PSets come with these measures precomputed, and accessible using the <code>sensitivityProfiles</code> function:</p>
<div class="sourceCode" id="cb78"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">sensitivityProfiles</span>(<span class="no">CCLE</span>)[,<span class="fl">1</span>:<span class="fl">5</span>]))</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left"></th>
<th align="right">ic50_published</th>
<th align="right">aac_published</th>
<th align="right">amax_published</th>
<th align="right">aac_recomputed</th>
<th align="right">ic50_recomputed</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">drugid_AEW541_1321N1</td>
<td align="right">8.000000</td>
<td align="right">0.0873750</td>
<td align="right">-42.55801</td>
<td align="right">0.0980082</td>
<td align="right">16.332263</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_22Rv1</td>
<td align="right">2.329924</td>
<td align="right">0.2205000</td>
<td align="right">-71.58934</td>
<td align="right">0.2132382</td>
<td align="right">2.034710</td>
</tr>
<tr class="odd">
<td align="left">drugid_AEW541_42-MG-BA</td>
<td align="right">2.682130</td>
<td align="right">0.1144375</td>
<td align="right">-63.49137</td>
<td align="right">0.1215116</td>
<td align="right">2.687672</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_5637</td>
<td align="right">5.002314</td>
<td align="right">0.1243550</td>
<td align="right">-62.35278</td>
<td align="right">0.1087770</td>
<td align="right">5.192732</td>
</tr>
<tr class="odd">
<td align="left">drugid_AEW541_639-V</td>
<td align="right">1.736181</td>
<td align="right">0.1936250</td>
<td align="right">-51.95981</td>
<td align="right">0.1834458</td>
<td align="right">5.113119</td>
</tr>
<tr class="even">
<td align="left">drugid_AEW541_697</td>
<td align="right">4.260822</td>
<td align="right">0.1087625</td>
<td align="right">-73.33379</td>
<td align="right">0.1051855</td>
<td align="right">4.254253</td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="radiogx-1" class="section level2">
<h2 class="hasAnchor">
<a href="#radiogx-1" class="anchor"></a>RadioGx</h2>
<div id="fitting-linear-quadratic-lq-models" class="section level3">
<h3 class="hasAnchor">
<a href="#fitting-linear-quadratic-lq-models" class="anchor"></a>Fitting Linear Quadratic (LQ) Models</h3>
<p>RadioGx provides a number of functions for analyzing dose response experiments. To use these functions, we must first fit a statistical model to the dose response data. Due to the nature of radiation treatment, the model used is different from the model in Pharmacogenomics. While Pharmacogenomics uses a log-logistic Hill Curve for modeling the dose-response relationship, Radiogenomics analysis traditionally uses a Linear Quadratic model. This is due to the different action of radiation treatment compared to compound inhibition of targets.</p>
<p>This package exports a function for fitting linear-quadratic models to dose response data. The function can be used with data contained in a RadioSet or with raw dose-response data.</p>
<div class="sourceCode" id="cb79"><html><body><pre class="r"><span class="co"># Extract raw sensitvity data from the RadioSet</span>
<span class="no">sensRaw</span> <span class="kw">&lt;-</span> <span class="fu">sensitivityRaw</span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<div class="sourceCode" id="cb80"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">sensRaw</span>)</pre></body></html></div>
<pre><code>##  num [1:305, 1:9, 1:2] 1 1 1 1 1 1 1 1 1 1 ...
##  - attr(*, "dimnames")=List of 3
##   ..$ : chr [1:305] "SK-N-FI_radiation_1" "SK-N-AS_radiation_3" "CHP-212_radiation_4" "MHH-NB-11_radiation_6" ...
##   ..$ : chr [1:9] "doses1" "doses2" "doses3" "doses4" ...
##   ..$ : chr [1:2] "Dose" "Viability"</code></pre>
<p>The data returned by <code>sensitivityRaw(RSet)</code> is a three dimensional array, but it can also be thought of as a set of experiment by treatment matrices. We can see by the <code>dimnames</code> of the third dimensions that the first matrix holds the radiation dose (in Gy) for each experiment, while the second matrix holds the viability measurements for the cell-line after each dose in the experimental series.</p>
<div class="sourceCode" id="cb82"><html><body><pre class="r"><span class="co"># Find a cancer cell-line of interest</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">sensitivityInfo</span>(<span class="no">Cleveland</span>)$<span class="no">cellid</span>)</pre></body></html></div>
<pre><code>## [1] "SK-N-FI"   "SK-N-AS"   "CHP-212"   "MHH-NB-11" "SK-N-SH"   "NB1"</code></pre>
<div class="sourceCode" id="cb84"><html><body><pre class="r"><span class="no">cancerCellLine</span> <span class="kw">&lt;-</span> <span class="fu">sensitivityInfo</span>(<span class="no">Cleveland</span>)$<span class="no">cellid</span>[<span class="fl">1</span>]
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">cancerCellLine</span>)</pre></body></html></div>
<pre><code>## [1] "SK-N-FI"</code></pre>
<div class="sourceCode" id="cb86"><html><body><pre class="r"><span class="co"># Get the radiation doses and associated survival data from clevelandSmall</span>
<span class="no">radiationDoses</span> <span class="kw">&lt;-</span> <span class="no">sensRaw</span>[<span class="fl">1</span>, , <span class="st">'Dose'</span>]
<span class="no">survivalFractions</span> <span class="kw">&lt;-</span> <span class="no">sensRaw</span>[<span class="fl">1</span>, , <span class="st">'Viability'</span>]</pre></body></html></div>
<div class="sourceCode" id="cb87"><html><body><pre class="r"><span class="no">LQmodel</span> <span class="kw">&lt;-</span> <span class="fu">linearQuadraticModel</span>(<span class="kw">D</span><span class="kw">=</span><span class="no">radiationDoses</span>,
                                <span class="kw">SF</span><span class="kw">=</span><span class="no">survivalFractions</span>)
<span class="no">LQmodel</span></pre></body></html></div>
<pre><code>##      alpha       beta 
## 0.15219719 0.01783452 
## attr(,"Rsquare")
## [1] 0.9882769</code></pre>
<p>Above we see that LQmodel contains the alpha and beta coefficients for the dose response curve fit to the dose and viability data for the IMR-32 cancer cell-line. Based on the <span class="math inline">\(R^2\)</span> attribute we can see that the model fit for this data is good, with 98.8% of observed variance explained by the model.</p>
</div>
<div id="calculating-dose-response-metrics" class="section level3">
<h3 class="hasAnchor">
<a href="#calculating-dose-response-metrics" class="anchor"></a>Calculating Dose-Response Metrics</h3>
<p>RadioGx provides a number of functions for calculating common dose response metrics such as surviving fraction (SF), area under the curve (AUC) and dose at which only 10% of cancer cells survive (D10).</p>
<p>Some of these functions require the alpha and beta coefficients, as calculated above using the <code>linearQuadraticModel</code> function.</p>
<div class="sourceCode" id="cb89"><html><body><pre class="r"><span class="no">survFracAfter2Units</span> <span class="kw">&lt;-</span> <span class="fu">computeSF2</span>(<span class="kw">pars</span><span class="kw">=</span><span class="no">LQmodel</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">survFracAfter2Units</span>)</pre></body></html></div>
<pre><code>## [1] 0.6867861</code></pre>
<div class="sourceCode" id="cb91"><html><body><pre class="r"><span class="no">dose10PercentSurv</span> <span class="kw">&lt;-</span> <span class="fu">computeD10</span>(<span class="kw">pars</span><span class="kw">=</span><span class="no">LQmodel</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">dose10PercentSurv</span>)</pre></body></html></div>
<pre><code>## [1] 7.870413</code></pre>
<p>We see from the above code cell that after two units of radiation, 68.679% of cancer cells remain relative to the initial population. Conversely, using <code>computeD10</code> we see that on average 7.87 units of radiation need to be administered to result in 10% cell-line survival (i.e., 90% of cancer cells are killed).</p>
<p>Other dose-response metrics can be computed directly using radiation dose and cancer cell viability data.</p>
<div class="sourceCode" id="cb93"><html><body><pre class="r"><span class="no">areaUnderDoseRespCurve</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeAUC.html">computeAUC</a></span>(<span class="kw">D</span><span class="kw">=</span><span class="no">radiationDoses</span>, <span class="kw">pars</span><span class="kw">=</span><span class="no">LQmodel</span>, <span class="kw">lower</span><span class="kw">=</span><span class="fl">0</span>,
                                     <span class="kw">upper</span><span class="kw">=</span><span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">areaUnderDoseRespCurve</span>)</pre></body></html></div>
<pre><code>## [1] 2.872288</code></pre>
<p>In the above code block we compute the AUC for a dose-response curve between a dose of 0 to 1 Gy. This area can be interpreted as the total proportion of cells killed during the administration of 1 Gy of radiation.</p>
</div>
<div id="dose-response-curves" class="section level3">
<h3 class="hasAnchor">
<a href="#dose-response-curves" class="anchor"></a>Dose-Response Curves</h3>
<p>The <code>doseResponseCurve</code> function can be used to generate plots of surviving fraction vs dose for radiation sensitivity experiments. In this example we provide raw data values to create the plot. When the <code>plot.type</code> is set to “Both”, a linear-quadratic model will also be fit to the supplied dose-response values.</p>
<div class="sourceCode" id="cb95"><html><body><pre class="r"><span class="fu">doseResponseCurve</span>(
  <span class="kw">Ds</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Experiment 1"</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">6</span>)),
  <span class="kw">SFs</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Experiment 1"</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">.6</span>,<span class="fl">.4</span>,<span class="fl">.2</span>)),
  <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Both"</span>
)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/plotting_rad_dose_resp-1.png" width="700"></p>
<p>Additionally, <code>doseResponseCurve</code> can be used to create dose response curves directly from a curated RadioSet object. When utilizing this feature, a cell-line must be selected from the RadioSet. This can be done by name if you know which cell-line you are looking for. If you don’t know which cell-line you want to visualize, the available cell-lines can be explored using the <code>cellInfo</code> function.</p>
<div class="sourceCode" id="cb96"><html><body><pre class="r"><span class="fu">doseResponseCurve</span>(
  <span class="kw">rSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">Cleveland</span>),
  <span class="kw">cellline</span><span class="kw">=</span><span class="fu">cellInfo</span>(<span class="no">Cleveland</span>)$<span class="no">cellid</span>[<span class="fl">3</span>]
)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/plotting_rad_dose_resp_rSet-1.png" width="700"></p>
</div>
</div>
<div id="xeva-2" class="section level2">
<h2 class="hasAnchor">
<a href="#xeva-2" class="anchor"></a>Xeva</h2>
<p>Drug response in PDX models is usually not assessed in the same way as <em>in vitro</em>. Rather than looking at viability vs dose curves, one looks at tumour growth vs time curves, to see if the treatement with a drug can inhibit tumour growth or initiate tumour shrinkage and clearance. Given the increased complexity of a tumour growing within a living mouse, there is no canonical model that is fit to this data. Rather, there are several options for assessing response in these models, each with its pros and cons.</p>
<div id="visualizing-tumour-growth-curves" class="section level3">
<h3 class="hasAnchor">
<a href="#visualizing-tumour-growth-curves" class="anchor"></a>Visualizing Tumour Growth Curves</h3>
<p>Xeva provides functions to visualize these PDX time vs. tumor volume curve.</p>
<div class="sourceCode" id="cb97"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/plotPDX.html">plotPDX</a></span>(<span class="kw">object</span> <span class="kw">=</span> <span class="no">brca</span>, <span class="kw">patient.id</span> <span class="kw">=</span> <span class="st">"X-1004"</span>, <span class="kw">drug</span> <span class="kw">=</span> <span class="st">"BGJ398"</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/1X3-1.png" width="700"></p>
<div class="sourceCode" id="cb98"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/plotPDX.html">plotPDX</a></span>(<span class="kw">object</span><span class="kw">=</span><span class="no">brca</span>,<span class="kw">patient.id</span><span class="kw">=</span><span class="st">"X-1004"</span>,<span class="kw">drug</span><span class="kw">=</span><span class="st">"BGJ398"</span>,<span class="kw">control.name</span><span class="kw">=</span><span class="st">"untreated"</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/1X3-2.png" width="700"></p>
<p>We can also plot the normalized volume.</p>
<div class="sourceCode" id="cb99"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/plotPDX.html">plotPDX</a></span>(<span class="kw">object</span><span class="kw">=</span><span class="no">brca</span>,<span class="kw">patient.id</span><span class="kw">=</span><span class="st">"X-1004"</span>,<span class="kw">drug</span><span class="kw">=</span><span class="st">"BGJ398"</span>,<span class="kw">control.name</span><span class="kw">=</span><span class="st">"untreated"</span>,
        <span class="kw">vol.normal</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/1X4-1.png" width="700"></p>
<p>When replicates are avaliable Xeva can be used the visualize each model.</p>
<div class="sourceCode" id="cb100"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"repdx"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/plotPDX.html">plotPDX</a></span>(<span class="kw">object</span><span class="kw">=</span><span class="no">repdx</span>,<span class="kw">patient.id</span><span class="kw">=</span><span class="st">"P1"</span>,<span class="kw">drug</span><span class="kw">=</span><span class="st">"treatment"</span>,<span class="kw">control.name</span><span class="kw">=</span><span class="st">"Control"</span>,
        <span class="kw">vol.normal</span> <span class="kw">=</span> <span class="fl">TRUE</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/1X5-1.png" width="700"></p>
</div>
<div id="quantifying-pdx-drug-response" class="section level3">
<h3 class="hasAnchor">
<a href="#quantifying-pdx-drug-response" class="anchor"></a>Quantifying PDX Drug Response</h3>
<p>In Xeva several response computing functions such as mRECIST, slope, AUC etc. have been implemented. Drug response can be computed for PDX models as:</p>
<div class="sourceCode" id="cb101"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/response.html">response</a></span>(<span class="no">brca</span>, <span class="kw">model.id</span><span class="kw">=</span><span class="st">"X.1004.BG98"</span>, <span class="kw">res.measure</span><span class="kw">=</span><span class="st">"mRECIST"</span>)</pre></body></html></div>
<pre><code>## computing mRECIST for X.1004.BG98</code></pre>
<pre><code>## $volume.change
## [1]   0.00000  -8.91337 -13.52028 -35.10265 -54.28142 -41.36204 -38.55784
## [8] -32.69905 -28.24236
## 
## $average.response
## [1]   0.000000  -4.456685  -7.477883 -14.384076 -22.363545 -25.529962 -27.391087
## [8] -28.054582 -28.075447
## 
## $best.response
## [1] -54.28142
## 
## $best.response.time
## [1] 12
## 
## $best.average.response
## [1] -28.07545
## 
## $best.average.response.time
## [1] 26
## 
## $mRECIST
## [1] "PR"</code></pre>
<p>Drug response for all models is stored in the sensitivity slot and can be extracted using</p>
<div class="sourceCode" id="cb104"><html><body><pre class="r"><span class="no">brca.mr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/summarizeResponse.html">summarizeResponse</a></span>(<span class="no">brca</span>, <span class="kw">response.measure</span> <span class="kw">=</span> <span class="st">"mRECIST"</span>,
                             <span class="kw">group.by</span><span class="kw">=</span><span class="st">"patient.id"</span>)</pre></body></html></div>
<p>We can visualize this matrix as:</p>
<div class="sourceCode" id="cb105"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/plotmRECIST.html">plotmRECIST</a></span>(<span class="no">brca.mr</span>, <span class="kw">control.name</span> <span class="kw">=</span> <span class="st">"untreated"</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/1X8-1.png" width="700"></p>
</div>
</div>
</div>
<div id="summary-functions" class="section level1">
<h1 class="hasAnchor">
<a href="#summary-functions" class="anchor"></a>Summary Functions</h1>
<div id="pharmacogx-1" class="section level2">
<h2 class="hasAnchor">
<a href="#pharmacogx-1" class="anchor"></a>PharmacoGx</h2>
<p>Pharmacogenomics studies often contain many examples of either replicated or missing data. One of the most common tasks in preparing data for statistical or machine learning analysis is aligning your features and labels. To accelerate using Pharmacogenomics data for analysis, PharmacoGx contains two functions which create deduplicated matrices with missing data filled by NAs: <code>summarizeSensitivityProfiles</code> and <code>summarizeMolecularProfiles</code>. They create matrices which are drugs x cell lines and molecular features x cell lines, with any cell lines profiled for only sensitivity or molecular features padded with NA values. For the molecular profiles, the data are returned in a <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> object, while drug sensitivity data are returned as a matrix.</p>
<div class="sourceCode" id="cb106"><html><body><pre class="r"><span class="no">mol.prof</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu">summarizeMolecularProfiles</span>(<span class="no">CCLE</span>, <span class="st">"rna"</span>)
<span class="no">sens.prof</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu">summarizeSensitivityProfiles</span>(<span class="no">CCLE</span>, <span class="st">"aac_recomputed"</span>)</pre></body></html></div>
<div class="sourceCode" id="cb107"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">mol.prof</span>)</pre></body></html></div>
<pre><code>## [1] 10012  1094</code></pre>
<div class="sourceCode" id="cb109"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="no">sens.prof</span>)</pre></body></html></div>
<pre><code>## [1]   24 1094</code></pre>
<p>Below, we use the summarizeSensitivityProfiles function and ggplot to investigate the distributions of AUC values within CCLE.</p>
<div class="sourceCode" id="cb111"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>, <span class="kw">verbose</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<pre><code>## Warning: package 'ggplot2' was built under R version 4.0.1</code></pre>
<div class="sourceCode" id="cb113"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">reshape2</span>, <span class="kw">verbose</span><span class="kw">=</span><span class="fl">FALSE</span>)
<span class="no">CCLE.aac</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu">summarizeSensitivityProfiles</span>(<span class="no">CCLE</span>, <span class="kw">sensitivity.measure</span> <span class="kw">=</span> <span class="st">"aac_recomputed"</span>)
<span class="no">melted_data</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span>(<span class="no">CCLE.aac</span>)
<span class="no">NA_rows</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">melted_data</span>), <span class="kw">arr.ind</span><span class="kw">=</span><span class="no">T</span>)[,<span class="fl">1</span>])
<span class="no">melted_data</span> <span class="kw">&lt;-</span> <span class="no">melted_data</span>[-<span class="no">NA_rows</span>,]
<span class="no">p</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">melted_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">Var1</span>,<span class="kw">y</span><span class="kw">=</span><span class="no">value</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span>(<span class="kw">fill</span><span class="kw">=</span><span class="st">"gray"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">axis.text.x</span><span class="kw">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">angle</span><span class="kw">=</span><span class="fl">90</span>,<span class="kw">hjust</span><span class="kw">=</span><span class="fl">1</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"Drugs"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"AAC"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<div class="figure">
<img src="GxWorkshop_files/figure-html/ccleaac-1.png" alt="Cells response to drugs in CCLE" width="700"><p class="caption">
Cells response to drugs in CCLE
</p>
</div>
</div>
<div id="radiogx-2" class="section level2">
<h2 class="hasAnchor">
<a href="#radiogx-2" class="anchor"></a>RadioGx</h2>
<div id="summarizing-sensitivity" class="section level3">
<h3 class="hasAnchor">
<a href="#summarizing-sensitivity" class="anchor"></a>Summarizing Sensitivity</h3>
<p>To retrieve a radiation type by cell-line summary of a sensitivity experiment, we use the <code>summarizeSensitivityProfiles</code> function. This will return a <code>matrix</code> where rows are radiation type<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>, columns are cell-line and values are viability measurements summarized using <code>summary.stat</code><a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>. The sensitivity measure to summarize can be specified using <code>sensitivity.measure</code><a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>.</p>
<div class="sourceCode" id="cb114"><html><body><pre class="r"><span class="no">sensSummary</span> <span class="kw">&lt;-</span> <span class="kw pkg">RadioGx</span><span class="kw ns">::</span><span class="fu">summarizeSensitivityProfiles</span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<div class="sourceCode" id="cb115"><html><body><pre class="r"><span class="no">sensSummary</span>[, <span class="fl">1</span>:<span class="fl">3</span>]</pre></body></html></div>
<pre><code>##  SK-N-FI  SK-N-AS  CHP-212 
## 2.861642 3.258931 1.344093</code></pre>
</div>
<div id="summarizing-molecular-data" class="section level3">
<h3 class="hasAnchor">
<a href="#summarizing-molecular-data" class="anchor"></a>Summarizing Molecular Data</h3>
<div class="sourceCode" id="cb117"><html><body><pre class="r"><span class="no">mprofSummary</span> <span class="kw">&lt;-</span> <span class="kw pkg">RadioGx</span><span class="kw ns">::</span><span class="fu">summarizeMolecularProfiles</span>(<span class="no">Cleveland</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">'rna'</span>, <span class="kw">summary.stat</span><span class="kw">=</span><span class="st">'median'</span>, <span class="kw">fill.missing</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb118"><html><body><pre class="r"><span class="no">mprofSummary</span></pre></body></html></div>
<pre><code>## class: SummarizedExperiment 
## dim: 20049 292 
## metadata(3): experimentData annotation protocolData
## assays(2): exprs se.exprs
## rownames(20049): ENSG00000000003 ENSG00000000005 ... ENSG00000280439
##   ENSG00000280448
## rowData names(7): Probe EnsemblGeneId ... BEST rownames
## colnames(292): SK-N-FI SK-N-AS ... VM-CUB-1 T.T
## colData names(25): samplename filename ... rownames tissueid</code></pre>
<p>Due to a lack of replicates in the Cleveland <code>RSet</code>, the returned <code>SummarizedExperiment</code> object contains the same information as the original. For other experiments with replicates, however, the result should contain one column per unique cell-line id. For ease of interoperability with the response data contained in an <code>RSet</code>, if <code>fill.missing</code> is FALSE empty columns for the cell-lines in the sensitivity experiment, but not in the molecular profile will be added to the <code>SummarizedExperiment</code> such that the dimensions are equal.</p>
</div>
</div>
<div id="xeva-3" class="section level2">
<h2 class="hasAnchor">
<a href="#xeva-3" class="anchor"></a>Xeva</h2>
<div id="summarizing-molecular-data-1" class="section level3">
<h3 class="hasAnchor">
<a href="#summarizing-molecular-data-1" class="anchor"></a>Summarizing Molecular Data</h3>
<p>As already seen above, Xeva provides a method to extract drug response data from XevaSets. Like PharmacoGx and RadioGx, the package also provides a method to extract the molecular data associated with a particular set of mice. For example, we can extract the RNAseq data for all the PDXs treated with trastuzumab, as below:</p>
<div class="sourceCode" id="cb120"><html><body><pre class="r"><span class="no">brca_trastuz</span> <span class="kw">&lt;-</span> <span class="kw pkg">Xeva</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/summarizeMolecularProfiles.html">summarizeMolecularProfiles</a></span>(<span class="no">brca</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">"RNASeq"</span>, <span class="kw">drug</span> <span class="kw">=</span> <span class="st">"trastuzumab"</span>)</pre></body></html></div>
</div>
</div>
</div>
<div id="signature-generation-and-biomarker-discovery" class="section level1">
<h1 class="hasAnchor">
<a href="#signature-generation-and-biomarker-discovery" class="anchor"></a>Signature Generation and Biomarker Discovery</h1>
<div id="drug-sensitivity-signatures" class="section level2">
<h2 class="hasAnchor">
<a href="#drug-sensitivity-signatures" class="anchor"></a>Drug Sensitivity Signatures</h2>
<p>Finally, we will talk about some functions built into PharmacoGx to generate signatures of molecular features which correlate with response to particular compounds. In PharmacoGx, the main function to generate signatures of response across drugs is <code>drugSensitivitySig</code>. It takes as input a PSet, and the molecular data type you wish to use to generate the signatures, and computes the linear association between each feature and each drug response vector independently. The default of using all features, drugs and cell lines in the dataset can be modified with parameters detailed below.</p>
<p>At its core, drugSensitivitySig fits a nested linear model to the drug response data, using the molecular feature, as well as cell tissue of origin and batch effect as possible predictors. The function then uses ANOVA to judge whether including the molecular feature in the model leads to a statistically significant improvement in prediction accuracy (R^2). For more details on the implementation, we encourage readers to reference the PharmacoGx manuscript, or the vignettes included in the package.</p>
<p>Here, as an example, we will run <code>drugSensitivitySig</code> to generate a signature of response to Lapatinib and Erlotinib, two targetted therapies, from the GDSC data. We will do this with a random 10 genes measured in the expression matrix, as the full computation can take some time to complete.</p>
<div class="sourceCode" id="cb121"><html><body><pre class="r"><span class="co">## Extracting the first 100 gene names. featureInfo brings up the rowData of the "rna" SummarizedExperiment</span>
<span class="no">chosen.genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sample.html">sample</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">featureInfo</span>(<span class="no">GDSC</span>, <span class="st">"rna"</span>), <span class="fl">10</span>))

<span class="no">sigs</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu">drugSensitivitySig</span>(<span class="no">GDSC</span>, <span class="st">"rna"</span>, <span class="kw">sensitivity.measure</span> <span class="kw">=</span> <span class="st">"aac_recomputed"</span>, <span class="kw">drugs</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Erlotinib"</span>, <span class="st">"Lapatinib"</span>), <span class="kw">features</span> <span class="kw">=</span> <span class="no">chosen.genes</span>)</pre></body></html></div>
<p>Lets look at the head of the signature for the two drugs:</p>
<div class="sourceCode" id="cb122"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">sigs</span>[,<span class="fl">1</span>,]))</pre></body></html></div>
<table class="table">
<colgroup>
<col width="18%">
<col width="12%">
<col width="11%">
<col width="4%">
<col width="12%">
<col width="11%">
<col width="11%">
<col width="4%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">estimate</th>
<th align="right">se</th>
<th align="right">n</th>
<th align="right">tstat</th>
<th align="right">fstat</th>
<th align="right">pvalue</th>
<th align="right">df</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000196700</td>
<td align="right">-0.0010362</td>
<td align="right">0.0549090</td>
<td align="right">288</td>
<td align="right">-0.0188716</td>
<td align="right">0.0003561</td>
<td align="right">0.9849579</td>
<td align="right">262</td>
<td align="right">0.9980546</td>
</tr>
<tr class="even">
<td align="left">ENSG00000168556</td>
<td align="right">-0.0916933</td>
<td align="right">0.0562915</td>
<td align="right">288</td>
<td align="right">-1.6289005</td>
<td align="right">2.6533169</td>
<td align="right">0.1045358</td>
<td align="right">262</td>
<td align="right">0.5675628</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000135378</td>
<td align="right">0.1108223</td>
<td align="right">0.0657545</td>
<td align="right">288</td>
<td align="right">1.6853948</td>
<td align="right">2.8405556</td>
<td align="right">0.0931030</td>
<td align="right">262</td>
<td align="right">0.5523516</td>
</tr>
<tr class="even">
<td align="left">ENSG00000162073</td>
<td align="right">-0.1035389</td>
<td align="right">0.0573381</td>
<td align="right">288</td>
<td align="right">-1.8057596</td>
<td align="right">3.2607678</td>
<td align="right">0.0721036</td>
<td align="right">262</td>
<td align="right">0.5037829</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000204291</td>
<td align="right">0.0232790</td>
<td align="right">0.0580488</td>
<td align="right">288</td>
<td align="right">0.4010251</td>
<td align="right">0.1608211</td>
<td align="right">0.6887285</td>
<td align="right">262</td>
<td align="right">0.9234843</td>
</tr>
<tr class="even">
<td align="left">ENSG00000177051</td>
<td align="right">-0.0765955</td>
<td align="right">0.0597004</td>
<td align="right">288</td>
<td align="right">-1.2829988</td>
<td align="right">1.6460860</td>
<td align="right">0.2006263</td>
<td align="right">262</td>
<td align="right">0.6892688</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb123"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="kw">x</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/as.data.frame.html">as.data.frame</a></span>(<span class="no">sigs</span>[,<span class="fl">2</span>,])))</pre></body></html></div>
<table class="table">
<colgroup>
<col width="18%">
<col width="12%">
<col width="11%">
<col width="4%">
<col width="12%">
<col width="11%">
<col width="11%">
<col width="4%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">estimate</th>
<th align="right">se</th>
<th align="right">n</th>
<th align="right">tstat</th>
<th align="right">fstat</th>
<th align="right">pvalue</th>
<th align="right">df</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000196700</td>
<td align="right">-0.0195924</td>
<td align="right">0.0563837</td>
<td align="right">297</td>
<td align="right">-0.3474839</td>
<td align="right">0.1207451</td>
<td align="right">0.7284975</td>
<td align="right">271</td>
<td align="right">0.9338826</td>
</tr>
<tr class="even">
<td align="left">ENSG00000168556</td>
<td align="right">0.0298141</td>
<td align="right">0.0571843</td>
<td align="right">297</td>
<td align="right">0.5213681</td>
<td align="right">0.2718247</td>
<td align="right">0.6025362</td>
<td align="right">271</td>
<td align="right">0.8970796</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000135378</td>
<td align="right">0.1623048</td>
<td align="right">0.0659125</td>
<td align="right">297</td>
<td align="right">2.4624277</td>
<td align="right">6.0635501</td>
<td align="right">0.0144225</td>
<td align="right">271</td>
<td align="right">0.2732868</td>
</tr>
<tr class="even">
<td align="left">ENSG00000162073</td>
<td align="right">-0.1072619</td>
<td align="right">0.0575102</td>
<td align="right">297</td>
<td align="right">-1.8650923</td>
<td align="right">3.4785694</td>
<td align="right">0.0632491</td>
<td align="right">271</td>
<td align="right">0.4785530</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000204291</td>
<td align="right">-0.0437106</td>
<td align="right">0.0587418</td>
<td align="right">297</td>
<td align="right">-0.7441151</td>
<td align="right">0.5537073</td>
<td align="right">0.4574515</td>
<td align="right">271</td>
<td align="right">0.8278657</td>
</tr>
<tr class="even">
<td align="left">ENSG00000177051</td>
<td align="right">-0.0443149</td>
<td align="right">0.0607281</td>
<td align="right">297</td>
<td align="right">-0.7297257</td>
<td align="right">0.5324996</td>
<td align="right">0.4661881</td>
<td align="right">271</td>
<td align="right">0.8315600</td>
</tr>
</tbody>
</table>
</div>
<div id="radiation-sensitivity-signatures" class="section level2">
<h2 class="hasAnchor">
<a href="#radiation-sensitivity-signatures" class="anchor"></a>Radiation Sensitivity Signatures</h2>
<p>The true usefulness of the RadioGx package comes from the ability to determine gene signatures for a cell-line from a sensitivity experiment. Cell-lines of interest to a given researcher can be selected and a molecular signature computed which correlates specific molecular features with a given sensitivity profile. Using this method one could identify signatures associated with either radio-sensitivity or radio-resistance. Combining this signature with drug response gene signatures from <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em>, as will be done in the subsequent section, one can identify drugs which could augment the effectiveness of a given radiation signature. Perhaps more powerfully, drugs which target features associated with radio-resistance can be found, potentially synergistically increasing the overall effectiveness of the combined treatment. We must note however that this is an active area of research, and the examples provided are illustrative only. When conducting research in this field, care must be taken in choosing the right model systems, measures of sensitivity, and incorporating prior knowledge to differentiate promising hits from experimental artifacts.</p>
<div class="sourceCode" id="cb124"><html><body><pre class="r"><span class="no">protein_coding</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">featInfo</span>$<span class="no">GeneBioType</span> <span class="kw">%in%</span> <span class="st">"protein_coding"</span>)
<span class="no">radSensSig</span> <span class="kw">&lt;-</span> <span class="fu">radSensitivitySig</span>(<span class="no">Cleveland</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">'rna'</span>, <span class="kw">nthread</span><span class="kw">=</span><span class="fl">16</span>, <span class="kw">features</span><span class="kw">=</span><span class="fu">fNames</span>(<span class="no">Cleveland</span>, <span class="st">'rna'</span>)[<span class="no">protein_coding</span>])</pre></body></html></div>
<pre><code>## Computing radiation sensitivity signatures...</code></pre>
<div class="sourceCode" id="cb126"><html><body><pre class="r"><span class="co"># Convert to a data.frame</span>
<span class="no">radSensSigDF</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="no">radSensSig</span>@<span class="kw">.Data</span>[, , ], <span class="kw">row.names</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">radSensSig</span>))

<span class="co"># Order by estimated correlation</span>
<span class="no">radSensSigDF</span> <span class="kw">&lt;-</span> <span class="no">radSensSigDF</span>[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">radSensSigDF</span>$<span class="no">estimate</span>, <span class="kw">decreasing</span><span class="kw">=</span><span class="fl">TRUE</span>), ]

<span class="co"># Subset to only significant FDR</span>
<span class="no">radSensSigDF</span> <span class="kw">&lt;-</span> <span class="no">radSensSigDF</span>[<span class="no">radSensSigDF</span>$<span class="no">fdr</span> <span class="kw">&lt;</span> <span class="fl">0.05</span>, ]</pre></body></html></div>
<div class="sourceCode" id="cb127"><html><body><pre class="r"><span class="no">radSensGeneSig</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">radSensSigDF</span>, <span class="fl">10</span>)
<span class="no">radResistGeneSig</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/utils/head.html">tail</a></span>(<span class="no">radSensSigDF</span>[<span class="no">radSensSigDF</span>$<span class="no">fdr</span> <span class="kw">&lt;</span> <span class="fl">0.05</span>, ], <span class="fl">10</span>)</pre></body></html></div>
<div class="sourceCode" id="cb128"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">radSensGeneSig</span>, <span class="kw">caption</span><span class="kw">=</span><span class="st">"Top 10 Gene Markers of Radio-sensitivity in the Cleveland RSet"</span>)</pre></body></html></div>
<table class="table">
<caption>Top 10 Gene Markers of Radio-sensitivity in the Cleveland RSet</caption>
<thead><tr class="header">
<th align="left"></th>
<th align="right">estimate</th>
<th align="right">se</th>
<th align="right">n</th>
<th align="right">tstat</th>
<th align="right">fstat</th>
<th align="right">pvalue</th>
<th align="right">df</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000151117</td>
<td align="right">0.3976407</td>
<td align="right">0.0595970</td>
<td align="right">292</td>
<td align="right">6.672160</td>
<td align="right">44.51772</td>
<td align="right">0.00e+00</td>
<td align="right">269</td>
<td align="right">0.0000023</td>
</tr>
<tr class="even">
<td align="left">ENSG00000132746</td>
<td align="right">0.3523199</td>
<td align="right">0.0708910</td>
<td align="right">292</td>
<td align="right">4.969883</td>
<td align="right">24.69974</td>
<td align="right">1.20e-06</td>
<td align="right">269</td>
<td align="right">0.0035660</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000070731</td>
<td align="right">0.3286902</td>
<td align="right">0.0659831</td>
<td align="right">292</td>
<td align="right">4.981431</td>
<td align="right">24.81465</td>
<td align="right">1.10e-06</td>
<td align="right">269</td>
<td align="right">0.0035660</td>
</tr>
<tr class="even">
<td align="left">ENSG00000124664</td>
<td align="right">0.3165793</td>
<td align="right">0.0771910</td>
<td align="right">292</td>
<td align="right">4.101246</td>
<td align="right">16.82022</td>
<td align="right">5.45e-05</td>
<td align="right">269</td>
<td align="right">0.0139882</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000148513</td>
<td align="right">0.3104869</td>
<td align="right">0.0616191</td>
<td align="right">292</td>
<td align="right">5.038810</td>
<td align="right">25.38960</td>
<td align="right">9.00e-07</td>
<td align="right">269</td>
<td align="right">0.0035660</td>
</tr>
<tr class="even">
<td align="left">ENSG00000183888</td>
<td align="right">0.3100016</td>
<td align="right">0.0647328</td>
<td align="right">292</td>
<td align="right">4.788945</td>
<td align="right">22.93399</td>
<td align="right">2.80e-06</td>
<td align="right">269</td>
<td align="right">0.0040761</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000159763</td>
<td align="right">0.3055805</td>
<td align="right">0.0602806</td>
<td align="right">292</td>
<td align="right">5.069301</td>
<td align="right">25.69781</td>
<td align="right">7.00e-07</td>
<td align="right">269</td>
<td align="right">0.0035660</td>
</tr>
<tr class="even">
<td align="left">ENSG00000110484</td>
<td align="right">0.2972925</td>
<td align="right">0.0613169</td>
<td align="right">292</td>
<td align="right">4.848461</td>
<td align="right">23.50757</td>
<td align="right">2.10e-06</td>
<td align="right">269</td>
<td align="right">0.0040761</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000145817</td>
<td align="right">0.2946001</td>
<td align="right">0.0610280</td>
<td align="right">292</td>
<td align="right">4.827293</td>
<td align="right">23.30276</td>
<td align="right">2.30e-06</td>
<td align="right">269</td>
<td align="right">0.0040761</td>
</tr>
<tr class="even">
<td align="left">ENSG00000110171</td>
<td align="right">0.2937235</td>
<td align="right">0.0659120</td>
<td align="right">292</td>
<td align="right">4.456296</td>
<td align="right">19.85857</td>
<td align="right">1.22e-05</td>
<td align="right">269</td>
<td align="right">0.0081632</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb129"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">radResistGeneSig</span>, <span class="kw">caption</span><span class="kw">=</span><span class="st">"Top 10 Gene Markers of Radio-resistance in the Cleveland RSet"</span>)</pre></body></html></div>
<table class="table">
<caption>Top 10 Gene Markers of Radio-resistance in the Cleveland RSet</caption>
<colgroup>
<col width="19%">
<col width="13%">
<col width="12%">
<col width="4%">
<col width="12%">
<col width="10%">
<col width="10%">
<col width="4%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">estimate</th>
<th align="right">se</th>
<th align="right">n</th>
<th align="right">tstat</th>
<th align="right">fstat</th>
<th align="right">pvalue</th>
<th align="right">df</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000106346</td>
<td align="right">-0.2650079</td>
<td align="right">0.0583550</td>
<td align="right">292</td>
<td align="right">-4.541304</td>
<td align="right">20.62344</td>
<td align="right">8.40e-06</td>
<td align="right">269</td>
<td align="right">0.0068289</td>
</tr>
<tr class="even">
<td align="left">ENSG00000213160</td>
<td align="right">-0.2652584</td>
<td align="right">0.0620200</td>
<td align="right">292</td>
<td align="right">-4.276979</td>
<td align="right">18.29255</td>
<td align="right">2.63e-05</td>
<td align="right">269</td>
<td align="right">0.0109258</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000129351</td>
<td align="right">-0.2659001</td>
<td align="right">0.0564573</td>
<td align="right">292</td>
<td align="right">-4.709758</td>
<td align="right">22.18182</td>
<td align="right">4.00e-06</td>
<td align="right">269</td>
<td align="right">0.0049492</td>
</tr>
<tr class="even">
<td align="left">ENSG00000056277</td>
<td align="right">-0.2680220</td>
<td align="right">0.0584539</td>
<td align="right">292</td>
<td align="right">-4.585184</td>
<td align="right">21.02392</td>
<td align="right">7.00e-06</td>
<td align="right">269</td>
<td align="right">0.0066171</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000164167</td>
<td align="right">-0.2740782</td>
<td align="right">0.0568968</td>
<td align="right">292</td>
<td align="right">-4.817111</td>
<td align="right">23.20455</td>
<td align="right">2.40e-06</td>
<td align="right">269</td>
<td align="right">0.0040761</td>
</tr>
<tr class="even">
<td align="left">ENSG00000105255</td>
<td align="right">-0.2816045</td>
<td align="right">0.0680826</td>
<td align="right">292</td>
<td align="right">-4.136217</td>
<td align="right">17.10829</td>
<td align="right">4.72e-05</td>
<td align="right">269</td>
<td align="right">0.0132329</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000101974</td>
<td align="right">-0.2837507</td>
<td align="right">0.0573454</td>
<td align="right">292</td>
<td align="right">-4.948095</td>
<td align="right">24.48364</td>
<td align="right">1.30e-06</td>
<td align="right">269</td>
<td align="right">0.0035660</td>
</tr>
<tr class="even">
<td align="left">ENSG00000254726</td>
<td align="right">-0.2863624</td>
<td align="right">0.0605707</td>
<td align="right">292</td>
<td align="right">-4.727737</td>
<td align="right">22.35150</td>
<td align="right">3.70e-06</td>
<td align="right">269</td>
<td align="right">0.0049416</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000189007</td>
<td align="right">-0.2894647</td>
<td align="right">0.0625417</td>
<td align="right">292</td>
<td align="right">-4.628346</td>
<td align="right">21.42158</td>
<td align="right">5.70e-06</td>
<td align="right">269</td>
<td align="right">0.0062083</td>
</tr>
<tr class="even">
<td align="left">ENSG00000166582</td>
<td align="right">-0.3080795</td>
<td align="right">0.0642025</td>
<td align="right">292</td>
<td align="right">-4.798563</td>
<td align="right">23.02620</td>
<td align="right">2.70e-06</td>
<td align="right">269</td>
<td align="right">0.0040761</td>
</tr>
</tbody>
</table>
</div>
<div id="in-vivo-biomarker-discovery" class="section level2">
<h2 class="hasAnchor">
<a href="#in-vivo-biomarker-discovery" class="anchor"></a><em>in vivo</em> Biomarker Discovery</h2>
<p>For <em>in vivo</em> biomarker discovery we need the genomic data along with drug response. As discussed above, this can be extracted using the <code>summarizeMolecularProfiles</code> function. The output will be an <code>ExpressionSet</code> object which contains genomic data and drug response.</p>
<div class="sourceCode" id="cb130"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/message.html">suppressMessages</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Biobase</span>))
<span class="no">dr.mol</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/summarizeMolecularProfiles.html">summarizeMolecularProfiles</a></span>(<span class="no">brca</span>, <span class="kw">drug</span><span class="kw">=</span><span class="st">"BGJ398"</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">"RNASeq"</span>,
                                    <span class="kw">tissue</span><span class="kw">=</span> <span class="st">"BRCA"</span>,
                                    <span class="kw">sensitivity.measure</span><span class="kw">=</span><span class="st">"best.average.response"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">dr.mol</span>)[<span class="fl">1</span>:<span class="fl">5</span>, <span class="fl">1</span>:<span class="fl">3</span>]</pre></body></html></div>
<pre><code>##          X.1004.BG98 X.1008.BG98 X.1298.BG98
## A1BG      1.90689060  3.31759350  2.28095631
## A1BG-AS1  1.79908731  2.08746284  1.83995959
## A1CF      0.02856915  0.04264434  0.01435529
## A2LD1     2.55336050  0.85598970  2.04963077
## A2M       0.01435529  6.17332735  0.00000000</code></pre>
<div class="sourceCode" id="cb132"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">dr.mol</span>)[<span class="fl">1</span>:<span class="fl">5</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"model.id"</span>, <span class="st">"best.average.response"</span>)]</pre></body></html></div>
<pre><code>##                model.id best.average.response
## X.1004.BG98 X.1004.BG98             -28.07545
## X.1008.BG98 X.1008.BG98              14.56386
## X.1298.BG98 X.1298.BG98              13.79941
## X.1371.BG98 X.1371.BG98              49.50584
## X.1383.BG98 X.1383.BG98              10.26197</code></pre>
<p>The output can be used for further inquiry such as correlation analysis.</p>
<div class="sourceCode" id="cb134"><html><body><pre class="r"><span class="no">genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">dr.mol</span>))

<span class="no">.computeCorTestDF</span> <span class="kw">&lt;-</span> <span class="kw">function</span>(<span class="no">g</span>, <span class="no">dr.mol</span>) {
  <span class="no">cr</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/stats/cor.test.html">cor.test</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/exprs.html">exprs</a></span>(<span class="no">dr.mol</span>)[<span class="no">g</span>, ], <span class="fu"><a href="https://rdrr.io/pkg/Biobase/man/phenoData.html">pData</a></span>(<span class="no">dr.mol</span>)$<span class="no">best.average.response</span>)
  <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span>(<span class="kw">gene</span><span class="kw">=</span><span class="no">g</span>, <span class="kw">cor</span><span class="kw">=</span><span class="no">cr</span>$<span class="no">estimate</span>, <span class="kw">p</span><span class="kw">=</span><span class="no">cr</span>$<span class="no">p.value</span>))
}

<span class="no">corDataFrames</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">genes</span>, <span class="no">.computeCorTestDF</span>, <span class="kw">dr.mol</span><span class="kw">=</span><span class="no">dr.mol</span>)</pre></body></html></div>
<pre><code>## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero

## Warning in cor(x, y): the standard deviation is zero</code></pre>
<div class="sourceCode" id="cb136"><html><body><pre class="r"><span class="no">output</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/do.call.html">do.call</a></span>(<span class="no">rbind</span>, <span class="no">corDataFrames</span>)

<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">output</span>)</pre></body></html></div>
<pre><code>##          gene         cor          p
## cor      A1BG -0.04245902 0.80296154
## cor1 A1BG-AS1 -0.02801888 0.86924726
## cor2     A1CF  0.03423020 0.84059963
## cor3    A2LD1 -0.01251831 0.94138025
## cor4      A2M  0.06572518 0.69913588
## cor5    A2ML1 -0.40761117 0.01227567</code></pre>
<p>Below, we will use this functionality of Xeva to see if an association discovered <em>in vitro</em> holds up in an <em>in vivo</em> setting.</p>
</div>
</div>
<div id="integrative-analysis-across-packages-and-disciplines" class="section level1">
<h1 class="hasAnchor">
<a href="#integrative-analysis-across-packages-and-disciplines" class="anchor"></a>Integrative Analysis across Packages and Disciplines</h1>
<p>In this section, we will go through two short examples of analysis which integrates across the packages presented above, highlighting the use of these packages to explore hypotheses across domains.</p>
<div id="comparing-sensitivity-signatures-between-radiation-and-drug-response" class="section level2">
<h2 class="hasAnchor">
<a href="#comparing-sensitivity-signatures-between-radiation-and-drug-response" class="anchor"></a>Comparing Sensitivity Signatures between Radiation and Drug Response</h2>
<p>Both RadioGx and PharmacoGx allow you to compute signatures of how much each molecular feature correlates with response to treatment, in the <em>in vitro</em> model systems. A natural question to ask is how the signature for response to gamma radiation will compare to signatures of response to various drugs. This can be used to generate hypothesis for combination therapies, or to gain insight into mechanisms of action.</p>
<p>We will run through the steps necessary to generate these results below, but note that the example datasets included with this workshop are incomplete (due to file size limits for github hosted R packages), and that due to time constraints the signatures we examine will have been precomputed from the full objects. We encourage the reader to work through this example with the full dataset.</p>
<p>To start, we will load the packages, and the data we will be using.</p>
<div class="sourceCode" id="cb138"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">PharmacoGx</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">RadioGx</span>)

<span class="no">GDSC1</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/downloadPSet.html">downloadPSet</a></span>(<span class="st">"GDSC_2019(v1_8.0)"</span>)
<span class="no">Cleveland</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/downloadRSet.html">downloadRSet</a></span>(<span class="st">"Cleveland"</span>)</pre></body></html></div>
<p><em>In reality, we run the code below for the purpose of building this vignette:</em></p>
<div class="sourceCode" id="cb139"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"GDSC"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"Cleveland"</span>)</pre></body></html></div>
<p>Now, we will compute signatures of radiation sensitivity and drug sensitivity for a selected set of drugs: Lapatinib, Dabrafenib, Paclitaxel, and Cisplatin. These drugs were chosen to have a mix of targetted and broad action therapies, and a couple of different targets for the two groups.</p>
<div class="sourceCode" id="cb140"><html><body><pre class="r"><span class="no">ft.rad</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">featureInfo</span>(<span class="no">Cleveland</span>, <span class="st">"rna"</span>))[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu">featureInfo</span>(<span class="no">Cleveland</span>, <span class="st">"rna"</span>)$<span class="no">GeneBioType</span> <span class="kw">==</span> <span class="st">"protein_coding"</span>)]

<span class="no">ft.info</span> <span class="kw">&lt;-</span> <span class="fu">featureInfo</span>(<span class="no">GDSC</span>, <span class="st">"rna"</span>)
<span class="no">ft.drug</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">ft.info</span>)[<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="no">ft.info</span>$<span class="no">GeneBioType</span> <span class="kw">==</span> <span class="st">"protein_coding"</span>)]

<span class="no">drugs.selected</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Lapatinib"</span>, <span class="st">"Paclitaxel"</span>, <span class="st">"Cisplatin"</span>, <span class="st">"Dabrafenib"</span>)</pre></body></html></div>
<p>To generate the signatures, we would run the code below.</p>
<div class="sourceCode" id="cb141"><html><body><pre class="r"><span class="no">nthread</span> <span class="kw">&lt;-</span> <span class="fl">16</span>
<span class="no">radSig</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/radSensitivitySig.html">radSensitivitySig</a></span>(<span class="no">Cleveland</span>, <span class="kw">mDataType</span> <span class="kw">=</span> <span class="st">"rna"</span>, <span class="kw">features</span><span class="kw">=</span><span class="no">ft.rad</span>, <span class="kw">nthread</span><span class="kw">=</span><span class="no">nthread</span>)</pre></body></html></div>
<div class="sourceCode" id="cb142"><html><body><pre class="r"><span class="no">gdscSigs</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu">drugSensitivitySig</span>(<span class="no">GDSC</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">"rna"</span>, <span class="kw">sensitivity.measure</span><span class="kw">=</span><span class="st">"aac_recomputed"</span>, <span class="kw">drugs</span><span class="kw">=</span><span class="no">drugs.selected</span>, <span class="kw">features</span><span class="kw">=</span><span class="no">ft.drug</span>, <span class="kw">nthread</span><span class="kw">=</span><span class="no">nthread</span>)</pre></body></html></div>
<p>However, for the purpose of this vignette, we will preload the precomputed signatures in interest of computation time.</p>
<div class="sourceCode" id="cb143"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">gdscSigs</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">radSig</span>)

<span class="no">common.genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/sets.html">intersect</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">gdscSigs</span>), <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">radSig</span>))
<span class="no">radSig</span> <span class="kw">&lt;-</span> <span class="no">radSig</span>[<span class="no">common.genes</span>,,,<span class="kw">drop</span><span class="kw">=</span><span class="fl">FALSE</span>]
<span class="no">gdscSigs</span> <span class="kw">&lt;-</span> <span class="no">gdscSigs</span>[<span class="no">common.genes</span>,,]</pre></body></html></div>
<p>Now, we can go directly to comparing the signatures. For this, we will use a function called <code>connectivityScore</code>, which wraps around two methods. It provides an interface to GSEA, for connecting your sensitivity signature to a disease or pathway gene set. It also implements genome wide weighted correlation, allowing the assessment of signature similarity, with the weights derived from significance estimates for the association of each gene with drug response. Here, we will use the genome-wide correlation, comparing standardized coefficients from each signature, weighted by our confidence that the gene is informative to predict response.</p>
<div class="sourceCode" id="cb144"><html><body><pre class="r"><span class="no">connectivity.res</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span>(<span class="no">drugs.selected</span>, <span class="kw">function</span>(<span class="no">drug</span>){
  <span class="fu"><a href="https://rdrr.io/r/base/function.html">return</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/connectivityScore.html">connectivityScore</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">radSig</span>[,<span class="fl">1</span>,<span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"estimate"</span>, <span class="st">"pvalue"</span>)], <span class="kw">y</span><span class="kw">=</span><span class="no">gdscSigs</span>[,<span class="no">drug</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"estimate"</span>,<span class="st">"pvalue"</span>)], <span class="kw">method</span> <span class="kw">=</span> <span class="st">"gwc"</span>, <span class="kw">nperm</span> <span class="kw">=</span> <span class="fl">200</span>))
})

<span class="fu"><a href="https://rdrr.io/r/base/names.html">names</a></span>(<span class="no">connectivity.res</span>) <span class="kw">&lt;-</span> <span class="no">drugs.selected</span>

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">connectivity.res</span>)</pre></body></html></div>
<pre><code>## $Lapatinib
##       score           p 
## 0.498671804 0.009950249 
## 
## $Paclitaxel
##        score            p 
## -0.008382044  0.230000000 
## 
## $Cisplatin
##        score            p 
## -0.510548599  0.009950249 
## 
## $Dabrafenib
##       score           p 
## 0.069866152 0.009950249</code></pre>
<p>The score returned can be interpreted as a correlation coefficient, ranging from <span class="math inline">\(-1\)</span> to <span class="math inline">\(1\)</span>. A positive correlation suggests that the similar molecular features distinguish cells responding to lapatinib treatment as to radiation. Of interest here is the negative correlation between the radiation response signature and cisplatin response. Cisplatin is a commonly used radiosensitizing agent, which when used in conjungtion with ionizing radiation has been shown to improve treatment efficacy. The negative correlation of the signatures can be naively interpreted to predict that radiation and cisplatin would target different cell populations in a tumour. However, the picture is probably not so simple: there is evidence that the radiosensitizing properties of cisplatin arise from its impairment of the Non-Homologous End Joining DNA repair pathway, and that cells which lack the NHEJ pathways are already hypersensitive to radiation [<span class="citation">(Boeckman et al. 2005)</span>]. Therefore, its probably synergistic mechanisms, not intra-tumoural heterogeneity, that drive efficacy of the combination of treatments.</p>
</div>
<div id="translational-biomarker-discovery" class="section level2">
<h2 class="hasAnchor">
<a href="#translational-biomarker-discovery" class="anchor"></a>Translational Biomarker Discovery</h2>
<p>Here, we look at a simple example of cross-species validation of biomarkers which are “discovered” within an <em>in vitro</em> setting, and then can be reproduced <em>in vivo</em>. For this example, we will focus on breast cancer models, a well known drug Trastuzumab, also known as Herceptin. We will look for gene expression biomarkers for this drug.</p>
<p>For the <em>in vitro</em> data in PharmacoGx, we will be using the <code>GRAY</code> dataset, from the Oregon Health and Science University, as it is the largest collection of <em>in vitro</em> drug sensitivity data within breast cancer cell lines.</p>
<p>First, we load the data. As always, these datasets could be downloaded from within the package, but we preloaded them for interest of time.</p>
<div class="sourceCode" id="cb146"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">PharmacoGx</span>)
<span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">Xeva</span>)

<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"brca_full"</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="st">"GRAY"</span>)</pre></body></html></div>
<p>Then we will run the <code>drugSensitivitySig</code> functions in both PharmacoGx and Xeva. In PharmacoGx, we run the function across gene expression for all protein coding genes, which we filter as above using the <code>featureInfo</code> table. We will look at the top 5 most significant hits from this correlation analysis.</p>
<div class="sourceCode" id="cb147"><html><body><pre class="r"><span class="no">ft.info</span> <span class="kw">&lt;-</span> <span class="fu">featureInfo</span>(<span class="no">GRAY</span>, <span class="st">"Kallisto_0.46.1.rnaseq"</span>)
<span class="no">ft.gray</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">ft.info</span>)[<span class="no">ft.info</span>$<span class="no">gene_type</span> <span class="kw">==</span> <span class="st">"protein_coding"</span>]


<span class="no">Trastuzumab.sig</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu">drugSensitivitySig</span>(<span class="no">GRAY</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">"Kallisto_0.46.1.rnaseq"</span>, <span class="kw">drug</span><span class="kw">=</span><span class="st">"Trastuzumab"</span>, <span class="kw">sensitivity.measure</span><span class="kw">=</span><span class="st">"aac_recomputed"</span>, <span class="kw">feature</span><span class="kw">=</span><span class="no">ft.gray</span>)</pre></body></html></div>
<pre><code>## Summarizing Kallisto_0.46.1.rnaseq molecular data for:   GRAY</code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |==========                                                            |  14%
  |                                                                            
  |====================                                                  |  29%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |==================================================                    |  71%
  |                                                                            
  |============================================================          |  86%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code>## Computing drug sensitivity signatures...</code></pre>
<div class="sourceCode" id="cb151"><html><body><pre class="r"><span class="no">Trastuzumab.sig</span> <span class="kw">&lt;-</span> <span class="no">Trastuzumab.sig</span>[<span class="fu"><a href="https://rdrr.io/r/base/order.html">order</a></span>(<span class="no">Trastuzumab.sig</span>[,<span class="fl">1</span>,<span class="st">"pvalue"</span>]),<span class="fl">1</span>,]
<span class="no">to.run</span> <span class="kw">&lt;-</span> <span class="no">ft.info</span>[<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="no">Trastuzumab.sig</span>), <span class="kw">n</span><span class="kw">=</span><span class="fl">5</span>),<span class="st">"gene_name"</span>]
<span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="no">Trastuzumab.sig</span>))</pre></body></html></div>
<table class="table">
<colgroup>
<col width="22%">
<col width="13%">
<col width="11%">
<col width="3%">
<col width="11%">
<col width="10%">
<col width="10%">
<col width="3%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left"></th>
<th align="right">estimate</th>
<th align="right">se</th>
<th align="right">n</th>
<th align="right">tstat</th>
<th align="right">fstat</th>
<th align="right">pvalue</th>
<th align="right">df</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000125686.12</td>
<td align="right">0.7008297</td>
<td align="right">0.1348064</td>
<td align="right">30</td>
<td align="right">5.198786</td>
<td align="right">27.02737</td>
<td align="right">1.61e-05</td>
<td align="right">28</td>
<td align="right">0.1969284</td>
</tr>
<tr class="even">
<td align="left">ENSG00000185361.9</td>
<td align="right">-0.6864392</td>
<td align="right">0.1374255</td>
<td align="right">30</td>
<td align="right">-4.994993</td>
<td align="right">24.94996</td>
<td align="right">2.81e-05</td>
<td align="right">28</td>
<td align="right">0.1969284</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000108306.13</td>
<td align="right">0.6849731</td>
<td align="right">0.1376865</td>
<td align="right">30</td>
<td align="right">4.974876</td>
<td align="right">24.74939</td>
<td align="right">2.97e-05</td>
<td align="right">28</td>
<td align="right">0.1969284</td>
</tr>
<tr class="even">
<td align="left">ENSG00000143479.17</td>
<td align="right">-0.6655933</td>
<td align="right">0.1410402</td>
<td align="right">30</td>
<td align="right">-4.719174</td>
<td align="right">22.27060</td>
<td align="right">5.98e-05</td>
<td align="right">28</td>
<td align="right">0.2691641</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000204520.14</td>
<td align="right">-0.6619936</td>
<td align="right">0.1416440</td>
<td align="right">30</td>
<td align="right">-4.673644</td>
<td align="right">21.84295</td>
<td align="right">6.77e-05</td>
<td align="right">28</td>
<td align="right">0.2691641</td>
</tr>
<tr class="even">
<td align="left">ENSG00000237190.4</td>
<td align="right">-0.6507459</td>
<td align="right">0.1434934</td>
<td align="right">30</td>
<td align="right">-4.535023</td>
<td align="right">20.56644</td>
<td align="right">9.88e-05</td>
<td align="right">28</td>
<td align="right">0.2960071</td>
</tr>
</tbody>
</table>
<p>Below, we try to see whether these top 5 hits will replicate looking at in vivo data, using the breast cancer PDXs from the Novartis PDX encyclopedia.</p>
<div class="sourceCode" id="cb152"><html><body><pre class="r"><span class="no">Trastuzumab.in.vivo</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/Xeva/man/drugSensitivitySig.html">drugSensitivitySig</a></span>(<span class="no">brca_full</span>, <span class="st">"trastuzumab"</span>, <span class="kw">mDataType</span> <span class="kw">=</span> <span class="st">"RNASeq"</span>, <span class="kw">features</span><span class="kw">=</span><span class="no">to.run</span>, <span class="kw">sensitivity.measure</span> <span class="kw">=</span> <span class="st">"best.average.response"</span>)</pre></body></html></div>
<pre><code>## Running for drug trastuzumab</code></pre>
<div class="sourceCode" id="cb154"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">Trastuzumab.in.vivo</span>)</pre></body></html></div>
<table class="table">
<colgroup>
<col width="11%">
<col width="13%">
<col width="12%">
<col width="11%">
<col width="3%">
<col width="12%">
<col width="11%">
<col width="11%">
<col width="3%">
<col width="11%">
</colgroup>
<thead><tr class="header">
<th align="left">feature</th>
<th align="left">drug</th>
<th align="right">estimate</th>
<th align="right">se</th>
<th align="right">n</th>
<th align="right">tstat</th>
<th align="right">fstat</th>
<th align="right">pvalue</th>
<th align="right">df</th>
<th align="right">fdr</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">MED1</td>
<td align="left">trastuzumab</td>
<td align="right">0.0481869</td>
<td align="right">0.1688345</td>
<td align="right">37</td>
<td align="right">0.2854089</td>
<td align="right">0.0814582</td>
<td align="right">0.7770132</td>
<td align="right">35</td>
<td align="right">0.7770132</td>
</tr>
<tr class="even">
<td align="left">TNFAIP8L1</td>
<td align="left">trastuzumab</td>
<td align="right">-0.3855750</td>
<td align="right">0.1559608</td>
<td align="right">37</td>
<td align="right">-2.4722563</td>
<td align="right">6.1120510</td>
<td align="right">0.0184327</td>
<td align="right">35</td>
<td align="right">0.0921637</td>
</tr>
<tr class="odd">
<td align="left">FBXL20</td>
<td align="left">trastuzumab</td>
<td align="right">-0.1490449</td>
<td align="right">0.1671428</td>
<td align="right">37</td>
<td align="right">-0.8917218</td>
<td align="right">0.7951678</td>
<td align="right">0.3786318</td>
<td align="right">35</td>
<td align="right">0.4732897</td>
</tr>
<tr class="even">
<td align="left">DYRK3</td>
<td align="left">trastuzumab</td>
<td align="right">0.2337671</td>
<td align="right">0.1643475</td>
<td align="right">37</td>
<td align="right">1.4223954</td>
<td align="right">2.0232088</td>
<td align="right">0.1637632</td>
<td align="right">35</td>
<td align="right">0.2729386</td>
</tr>
<tr class="odd">
<td align="left">MICA</td>
<td align="left">trastuzumab</td>
<td align="right">0.2747796</td>
<td align="right">0.1625244</td>
<td align="right">37</td>
<td align="right">1.6906976</td>
<td align="right">2.8584585</td>
<td align="right">0.0997829</td>
<td align="right">35</td>
<td align="right">0.2494571</td>
</tr>
</tbody>
</table>
<p>As we can see, no biomarker quite reaches statistical significance in vivo, after correcting for multiple testing. Prior to the multiple testing correction, TNFAIP8L1 does reach significance. The analysis is limited in this case by sample size, as you can see the sample size <em>in vivo</em> is much smaller than the sample size <em>in vitro</em>. This is one of the major reasons to follow this path of analysis, where potential hits are ranked first using larger <em>in virto</em> datasets, prior to being tested either in publically available PDX data, or in a PDX based drug screen.</p>
<div id="refs" class="references">
<div id="ref-boeckman_cisplatin_2005">
<p>Boeckman, Heather J., Kelly S. Trego, Karen M. Henkels, and John J. Turchi. 2005. “Cisplatin Sensitizes Cancer Cells to Ionizing Radiation via Inhibition of Non-Homologous End Joining.” <em>Molecular Cancer Research : MCR</em> 3 (5): 277–85. <a href="https://doi.org/10.1158/1541-7786.MCR-04-0032">https://doi.org/10.1158/1541-7786.MCR-04-0032</a>.</p>
</div>
</div>
</div>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>exported by <em><a href="https://bioconductor.org/packages/3.12/CoreGx">CoreGx</a></em><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>microarray<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>Note there is currently only one radiation type available, but in the future there will be more<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>Available options are ‘mean’, ‘median’, ‘first’, ‘last’<a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>See <code>sensitivityMeasures(RSet)</code> for available options<a href="#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Petr Smirnov.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
