<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Biomarker Discovery from High Throughput Screening Datasets • gxbioc2020workshop</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="Biomarker Discovery from High Throughput Screening Datasets">
<meta property="og:description" content="gxbioc2020workshop">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><!-- Global site tag (gtag.js) - Google Analytics --><script async src="https://www.googletagmanager.com/gtag/js?id=UA-93043521-1"></script><script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'UA-93043521-1');
</script>
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">gxbioc2020workshop</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.2.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/GxWorkshop.html">Biomarker Discovery from High Throughput Screening Datasets</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/bhklab/bioc2020workshop">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="GxWorkshop_files/accessible-code-block-0.0.1/empty-anchor.js"></script><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>Biomarker Discovery from High Throughput Screening Datasets</h1>
                        <h4 class="author">Petr Smirnov</h4>
            <address class="author_afil">
      Princess Margaret Cancer Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:psmirnov2000@gmail.com" class="email">psmirnov2000@gmail.com</a>
      </address>
                              <h4 class="author">Arvind Mer</h4>
            <address class="author_afil">
      Princess Margaret Cancer Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:arvind.mer@uhnresearch.ca" class="email">arvind.mer@uhnresearch.ca</a>
      </address>
                              <h4 class="author">Christopher Eeles</h4>
            <address class="author_afil">
      Princess Margaret Cancer Centre<br><a class="author_email" href="mailto:#"></a><a href="mailto:christopher.eeles@uhnresearch.ca" class="email">christopher.eeles@uhnresearch.ca</a>
      </address>
                  
      
      <small class="dont-index">Source: <a href="https://github.com/bhklab/bioc2020workshop/blob/master/vignettes/GxWorkshop.Rmd"><code>vignettes/GxWorkshop.Rmd</code></a></small>
      <div class="hidden name"><code>GxWorkshop.Rmd</code></div>

    </div>

    
    
<!-- Remove {-} from header to show numbering -->
<div id="workshop-overview" class="section level1 unnumbered">
<h1 class="hasAnchor">
<a href="#workshop-overview" class="anchor"></a>Workshop Overview</h1>
<div id="instructors-names-and-contact-information" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#instructors-names-and-contact-information" class="anchor"></a>Instructor(s) name(s) and contact information</h2>
<ul>
<li>Petr Smirnov <a href="mailto:petr.smirnov@uhnresearch.ca" class="email">petr.smirnov@uhnresearch.ca</a>
</li>
<li>Arvind Mer <a href="mailto:arvind.mer@uhnresearch.ca" class="email">arvind.mer@uhnresearch.ca</a>
</li>
<li>Christopher Eeles <a href="mailto:christopher.eeles@uhnresearch.ca" class="email">christopher.eeles@uhnresearch.ca</a>
</li>
</ul>
</div>
<div id="workshop-description" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#workshop-description" class="anchor"></a>Workshop Description</h2>
<p>This workshop will focus on the synergies between analysis results from the <code>PharmacoGx</code>, <code>Xeva</code> and <code>RadioGx</code> packages and their usefulness for discovery of biomarkers of drug and/or radiation sensitivity in cancer cell lines (CCLs) and patent derived xenograft models (PDXs). We will discuss issues with data curation, consistency and reproducibility within the literature as well as illustrate the importance of unified analytical platforms, data and code sharing in bioinformatics and biomedical research. In this lab learners will be led through an analysis for each of the three packages on data provided by the download functions within them. The results of these analyses will then be explored to highlight how drug and radiation dose-response profiles in CCLs and PDXs can be used to discover potential synergistic biomarkers for drug-radiation and drug combination therapies. The resulting biomarkers will be discussed in the context of translational cancer research and clinical applications of genomic data. We will conclude with a discussion of how these biomarkers can be used to inform future in vitro and in vivo treatment screenings and ultimately provide useful insights for clinical trial design.</p>
<div id="pre-requisites" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#pre-requisites" class="anchor"></a>Pre-requisites</h3>
<p>List any workshop prerequisites:</p>
<ul>
<li>Basic knowledge of R syntax</li>
<li>Knowledge of or interest in pharmacogenomics and radio-genomics</li>
<li>Familiarity with base bioconductor objects, including the SummarizedExperiment Class</li>
<li>Familiarity with linear modelling in R; ANOVA, and goodness of fit tests</li>
</ul>
<p>List relevant background reading for the workshop, including any theoretical background you expect students to have.</p>
<ul>
<li>Smirnov, P. et al. PharmacoGx: An R package for analysis of large pharmacogenomic datasets. Bioinformatics 32, 1244–1246 (2016).</li>
<li>Manem, V. SK. et al. Modeling Cellular Response in Large-Scale Radiogenomic Databases to Advance Precision Radiotherapy. Cancer Research (2019) <a href="doi:10.1158/0008-5472.CAN-19-0179" class="uri">doi:10.1158/0008-5472.CAN-19-0179</a>.</li>
<li>Mer, A. S. et al. Integrative pharmacogenomics analysis of patient-derived xenografts. Cancer Res canres.0349.2019 (2019) <a href="doi:10.1158/0008-5472.CAN-19-0349" class="uri">doi:10.1158/0008-5472.CAN-19-0349</a>.</li>
</ul>
</div>
<div id="workshop-participation" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#workshop-participation" class="anchor"></a>Workshop Participation</h3>
<p>Participants expected to have the following required packages installed on their machines to be able to run the commands along with the instructors. <code>PharmacoGx</code>, <code>Xeva</code>, <code>RadioGx</code>, <code>CoreGx</code> and <code>Biobase</code> from Bioconductor The workshop will be presented as a set of analysis steps to be replicated by the learners, with instructors available to explain the why and how of applying these functions to the given datasets. Learners will write analysis scripts as well as use interactive commands to explore the generated data structures and results. Learners will then brainstorm potential applications of the analysis results in their field as well as comment on use case examples presented by the instructors on research in our lab.</p>
</div>
<div id="r-bioconductor-packages-used" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#r-bioconductor-packages-used" class="anchor"></a><em>R</em> / <em>Bioconductor</em> packages used</h3>
<p>Bioconductor:</p>
<ul>
<li><a href="https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html" class="uri">https://bioconductor.org/packages/release/bioc/html/PharmacoGx.html</a></li>
<li><a href="http://bioconductor.org/packages/release/bioc/html/Xeva.html" class="uri">http://bioconductor.org/packages/release/bioc/html/Xeva.html</a></li>
<li>
<a href="https://cran.r-project.org/web/packages/RadioGx/index.html" class="uri">https://cran.r-project.org/web/packages/RadioGx/index.html</a>
<ul>
<li>RadioGx is currently available in the devel version of Bioconductor</li>
</ul>
</li>
<li>
<a href="https://cran.r-project.org/web/packages/CoreGx/index.html" class="uri">https://cran.r-project.org/web/packages/CoreGx/index.html</a>
<ul>
<li>CoreGx is currently available in the 3.11 release of Bioconductor</li>
</ul>
</li>
</ul>
</div>
</div>
<div id="time-outline" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#time-outline" class="anchor"></a>Time outline</h2>
<p>For a 2-hr workshop:</p>
<table class="table">
<thead><tr class="header">
<th>Activity</th>
<th>Time</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>Introduction</td>
<td>5m</td>
</tr>
<tr class="even">
<td>Data Structure and Common Functions Between Packages</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Basic functionalities of PharmacoGx</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Basic functionalities of Xeva</td>
<td>10m</td>
</tr>
<tr class="odd">
<td>Basic functionalities of RadioGx</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Data Analysis Walkthrough</td>
<td>30m</td>
</tr>
<tr class="odd">
<td>Understanding Analysis Results and Synergies</td>
<td>10m</td>
</tr>
<tr class="even">
<td>Use Case Discussion: Translational Science</td>
<td>15m</td>
</tr>
</tbody>
</table>
</div>
<div id="workshop-goals-and-objectives" class="section level2 unnumbered">
<h2 class="hasAnchor">
<a href="#workshop-goals-and-objectives" class="anchor"></a>Workshop goals and objectives</h2>
<div id="learning-goals" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#learning-goals" class="anchor"></a>Learning goals</h3>
<ul>
<li>Describe pharmacogenomic datasets and radiogenomic datasets and usefulness in cancer research</li>
<li>Learn how to extract information from these datasets and to intersect them over their common measured features, experiments, model systems</li>
<li>Learn how to visualize experimental results from these datasets</li>
<li>Learn how to model dose-response for both small compound and radiotherapy datasets</li>
<li>Learn measures to quantify the response in both cell line and PDX sensitivity screens, for both drug and radiation treatments</li>
<li>Understand the differences in experimental design between radiotherapy and small compound treatments in cell lines</li>
<li>Understand the differences in measuring response to therapy between cell line and PDX based models</li>
<li>Understand how to choose the appropriate sensitivity quantification metric for your task</li>
<li>Understand how to identify potential genomic signatures of drug and radiation response and explore how synergies between these signatures can inform clinical trial design</li>
</ul>
</div>
<div id="learning-objectives" class="section level3 unnumbered">
<h3 class="hasAnchor">
<a href="#learning-objectives" class="anchor"></a>Learning objectives</h3>
<ul>
<li>List available standardized pharmacogenomic and radiogenomic datasets and download them</li>
<li>Access the molecular features, dose-response and metadata contained within the data structures defined in the packages</li>
<li>Create drug-dose and radiation-dose response plots for cell line experiments</li>
<li>Create tumor growth curve plots for PDX experiments</li>
<li>Fit linear-quadratic models to radiotherapy experiments in cell lines</li>
<li>Fit Hill Slope models to dose-response experiments using small compound treatments in cell lines</li>
<li>Calculate the AAC, AUC, IC50, SF2, D10 metrics for response quantification in cell lines</li>
<li>Calculate the mRECIST, Slope, ABC, linear mix model and other tumour growth curve summary metrics</li>
<li>Visualize response of PDX experiments using an Oncoprint heatmap</li>
<li>Predict in vivo and in vitro univariate biomarkers using the PharmacoGx, Xeva and RadioGx packages</li>
<li>Compare signatures of response between chemotherapy compounds and radiotherapy</li>
<li>Compare univariate biomarkers between cell line and PDX experiments</li>
</ul>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h1>
<p>This tutorial, titled Biomarker Discovery from High Throughput Screening Datasets, focuses on a suite of packages designed to simplify analysis correlating measures of treatment response measured on model systems with molecular profiles of those systems. These packages primarly focus on modeling response to cancer treatments in cancer models, but the concepts presented can be applied to other disease types, given availability of data for the area of interest.</p>
<p>The first two of the packages, <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em> and <code>r BiocStyle::Biocpkg("RadioGx")</code> focus on screens done using cell lines as a model system. PharmacoGx focuses on modeling and predicting the response of cell lines treated with small molecule therapies. RadioGx swaps the small molecule treatment for irradiation of cell lines. The third package, <code>r BiocStyle::Biocpkg("Xeva")</code>, looks at treatment response for <em>in vivo</em> systems, primarily Patient Derived Xenographs of cancer tumours growing in immune compromised mice.</p>
</div>
<div id="data-structure-and-common-functions-between-packages" class="section level1">
<h1 class="hasAnchor">
<a href="#data-structure-and-common-functions-between-packages" class="anchor"></a>Data Structure and Common Functions Between Packages</h1>
<p>While the experiments analyzed by the three packages are different, the workflow that these packages accommodate has many similarities between them. First and foremost, all three packages were developed to provide data structures that package together screening and molecular data, as well as all related metadata, for complete studies. These packages also provide an interface to download large, publically available datasets which have standardized naming conventions, so that data across different studies is immediately comparable.</p>
<p>Furthermore, the analysis workflow shares common tasks across the three domains covered by these packages. This includes understanding and visualizing the data that is present in each dataset; modeling the response of the model systems to treatment; summarizing that response into a measure comparable across models; extracting and summarizing measurements of molecular features; and ultimately correlating the response of model systems to their molecular state.</p>
<p>In this tutorial, we will be highlighting each analysis step across all three packages, and then finish with a demonstration of how to integrate across the different domains presented here.</p>
<div id="experimental-designs" class="section level2">
<h2 class="hasAnchor">
<a href="#experimental-designs" class="anchor"></a>Experimental Designs</h2>
<p>RadioGx and PharmacoGx are both used to analyze data from in vitro, cell based assays. In cancer, the most common in vitro model used is the cancer cell line. These cell lines are usually obtained from patient tumour biopsies, which are passaged in a dish and often treated with factors to immortalize the cells - that is, to grant them infinite proliferation capacity. Such models are often deposited in repositories such as the ATCC, and made available for purchase to researchers who wish to use them in their experiments.</p>
<p><img src="CellLineFigure500ppi.png"></p>
<div id="pharmacogenomics" class="section level3">
<h3 class="hasAnchor">
<a href="#pharmacogenomics" class="anchor"></a>Pharmacogenomics</h3>
<p>In Pharmacogenomics, the design of the experiment usually involves doing two things in parallel: Each cell line in the study is profiled on a molecular level, which can include measuring DNA level aberations such as SNVs and copy number changes, RNA expression levels, DNA methylation, and protein expression. In parallel, the cell lines undergo a viability based drug screen, where cell lines are treated with a panel of drugs, over a range of concentrations, and the growth of each cell line is compared to an untreated control.</p>
<p>The variable response to drug treatement across different cell lines is then compared with the molecular states of the cells, using machine learning or statistical approaches to discern which molecular features are predictive of the observed drug response.</p>
<p><img src="pharmacogx_design_exp.png"></p>
</div>
<div id="radiogenomics" class="section level3">
<h3 class="hasAnchor">
<a href="#radiogenomics" class="anchor"></a>Radiogenomics</h3>
<p>Radiogenomics follows a very similar design to Pharmacogenomics. The main difference is the form of the treatment applied to cells - in Pharmacogenomics, small molecule treatments are studied, while in radiogenomics, the treatement is irradiation of the cells. While the design of the study is similar, the difference in treatement causes differences in how the response of cells is modelled and quantified, as we detail below.</p>
<p>Currently, there is only one major, high-throughput dataset looking at <em>in vitro</em> radiogenomics, the Cleveland Clinic dataset. In this dataset, gamma radiation was used as the treatment. Conceptually however, there is no reason why alpha and beta radiation could not be studied in a similar experimental setup.</p>
<p><img src="radiogx_design_exp.png"></p>
<p>Xeva:</p>
<p><img src="PDXfigure500ppi.png"></p>
</div>
<div id="pharmacoset-structure" class="section level3">
<h3 class="hasAnchor">
<a href="#pharmacoset-structure" class="anchor"></a>PharmacoSet Structure</h3>
<p>The basic structure of a PharmacoSet is as follows:</p>
<p><img src="PSetStruct.png"></p>
</div>
<div id="downloading-pharmacoset-objects" class="section level3">
<h3 class="hasAnchor">
<a href="#downloading-pharmacoset-objects" class="anchor"></a>Downloading PharmacoSet objects</h3>
<p>Let us download the PSets used for this tutorial into our local working environment. A table of available PharmacoSet objects can be obtained by using the <em>availablePSets</em> function.</p>
<div class="sourceCode" id="cb1"><html><body><pre class="r">  <span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">PharmacoGx</span>)
  <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/availablePSets.html">availablePSets</a></span>()</pre></body></html></div>
<p>If you look through the table produced by availablePSets, you will notice that each PSet object is downloaded from Zenodo - an Open Science project supported by the European Commision and CERN. Each PSet is assigned a DOI, which can be used to uniquely identify the data used in your analysis and retrieve it in the future for research reproducibility.</p>
<p>Any of the PharmacoSets in the table can then be downloaded by calling <em>downloadPSet</em>, which saves the datasets into a directory of the users choice, and returns the data into the R session. Lets try downloading the GDSC and CCLE PSets. The following code is included as an example, but not run, as the objects were preloaded onto the VM for the workshop.</p>
<div class="sourceCode" id="cb2"><html><body><pre class="r">  <span class="no">GDSC</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/downloadPSet.html">downloadPSet</a></span>(<span class="st">"GDSC_2020(v1-8.2)"</span>, <span class="kw">saveDir</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"."</span>, <span class="st">"PSets"</span>))
  <span class="no">CCLE</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/downloadPSet.html">downloadPSet</a></span>(<span class="st">"CCLE_2015"</span>, <span class="kw">saveDir</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/file.path.html">file.path</a></span>(<span class="st">"."</span>, <span class="st">"PSets"</span>))</pre></body></html></div>
<p>To load the datasets used in this workshop, we run the following code:</p>
<div class="sourceCode" id="cb3"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">GDSC</span>)
<span class="fu"><a href="https://rdrr.io/r/utils/data.html">data</a></span>(<span class="no">CCLE</span>)</pre></body></html></div>
</div>
<div id="examining-and-extracting-data-of-interest" class="section level3">
<h3 class="hasAnchor">
<a href="#examining-and-extracting-data-of-interest" class="anchor"></a>Examining and Extracting data of interest</h3>
<p>For most of your work with PSet objects, we recommend you use the accessor functions implemented in the package to access the data stored in the object. For example, here we can use the <em>cellInfo</em> function to pull out the tissue of origin for each cell line in the CCLE dataset. The <em>tissueid</em> column is standard for PSets, and should be present in all objects.</p>
<div class="sourceCode" id="cb4"><html><body><pre class="r"><span class="no">mycol</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"#8dd3c7"</span>,<span class="st">"#ffffb3"</span>,<span class="st">"#bebada"</span>,<span class="st">"#fb8072"</span>,<span class="st">"#80b1d3"</span>,<span class="st">"#fdb462"</span>,
           <span class="st">"#b3de69"</span>,<span class="st">"#fccde5"</span>,<span class="st">"#d9d9d9"</span>,<span class="st">"#bc80bd"</span>,<span class="st">"#ccebc5"</span>,<span class="st">"#ffed6f"</span>,
           <span class="st">"#a6cee3"</span>,<span class="st">"#1f78b4"</span>,<span class="st">"#b2df8a"</span>,<span class="st">"#33a02c"</span>,<span class="st">"#fb9a99"</span>,<span class="st">"#e31a1c"</span>,
           <span class="st">"#fdbf6f"</span>,<span class="st">"#ff7f00"</span>,<span class="st">"#cab2d6"</span>,<span class="st">"#6a3d9a"</span>,<span class="st">"#ffff99"</span>,<span class="st">"#b15928"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/pie.html">pie</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/table.html">table</a></span>(<span class="no">CCLE</span>@<span class="kw">cell</span>[,<span class="st">"tissueid"</span>]),
    <span class="kw">col</span><span class="kw">=</span><span class="no">mycol</span>,
    <span class="kw">main</span><span class="kw">=</span><span class="st">"Tissue types"</span>,
    <span class="kw">radius</span><span class="kw">=</span><span class="fl">1</span>,
    <span class="kw">cex</span><span class="kw">=</span><span class="fl">0.8</span>)</pre></body></html></div>
<div class="figure">
<img src="GxWorkshop_files/figure-html/piechart-1.png" alt="Tissue of origin of cell lines in CCLE study" width="700"><p class="caption">
Tissue of origin of cell lines in CCLE study
</p>
</div>
<p>While we will introduce some of the functions for subsetting and extracting data from PSets in this tutorial, the documentation of the PharmacoSet object contains descriptions of the available functions to interface with PSet objects:</p>
<div class="sourceCode" id="cb5"><html><body><pre class="r">?<span class="no">`PharmacoSet-class`</span></pre></body></html></div>
<p>PSets can be subsetted by refering directly to the drugs and cell lines you want to keep in the dataset. For example, we can subset the GDSC PSet in the following ways:</p>
<div class="sourceCode" id="cb6"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">GDSC</span>)</pre></body></html></div>
<pre><code>## Name:  GDSC_v1 
## Date Created:  Fri Jun  5 20:29:32 2020 
## Number of cell lines:  1064 
## RNA: 
##  Dim:  5947 789 
## Drug pertubation: 
##  Please look at pertNumber(cSet) to determine number of experiments for each drug-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  6890 
##  Please look at sensNumber(cSet) to determine number of experiments for each drug-cell combination.</code></pre>
<div class="sourceCode" id="cb8"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugNames.html">drugNames</a></span>(<span class="no">GDSC</span>)[<span class="fl">1</span>])</pre></body></html></div>
<pre><code>## [1] "Paclitaxel"</code></pre>
<div class="sourceCode" id="cb10"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/subsetTo.html">subsetTo</a></span>(<span class="no">GDSC</span>, <span class="kw">drugs</span><span class="kw">=</span><span class="st">"Erlotinib"</span>))</pre></body></html></div>
<pre><code>## Name:  GDSC_v1 
## Date Created:  Fri Jun  5 20:29:32 2020 
## Number of cell lines:  824 
## RNA: 
##  Dim:  5947 789 
## Drug pertubation: 
##  Please look at pertNumber(cSet) to determine number of experiments for each drug-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  511 
##  Please look at sensNumber(cSet) to determine number of experiments for each drug-cell combination.</code></pre>
<div class="sourceCode" id="cb12"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">GDSC</span>[<span class="st">"YT"</span>,<span class="st">"Erlotinib"</span>])</pre></body></html></div>
<pre><code>## Name:  GDSC_v1 
## Date Created:  Fri Jun  5 20:29:32 2020 
## Number of cell lines:  1 
## RNA: 
##  Dim:  5947 0 
## Drug pertubation: 
##  Please look at pertNumber(cSet) to determine number of experiments for each drug-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  1 
##  Please look at sensNumber(cSet) to determine number of experiments for each drug-cell combination.</code></pre>
<p>As an exercise, lets manually compute the number of overlaping cell lines and drugs between CCLE and GDSC. Try it yourself!</p>
<p>If you noticed as you completed the previous exercise, cell line and drug names are standardized between PSets downloaded using the <em>downloadPSet</em> function. This is the major benefit in working with the pre-curated objects: entities follow a standardized naming scheme, which was chosen to be human readable whenever possible.</p>
<p>PharmacoGx also implements a convenience function to intersect PSets:</p>
<div class="sourceCode" id="cb14"><html><body><pre class="r"><span class="no">common</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/intersectPSet.html">intersectPSet</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">intersectOn</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"drugs"</span>, <span class="st">"cell.lines"</span>))
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">common</span>)</pre></body></html></div>
</div>
<div id="working-with-dose-response-data-from-psets" class="section level3">
<h3 class="hasAnchor">
<a href="#working-with-dose-response-data-from-psets" class="anchor"></a>Working with Dose Response data from PSets</h3>
</div>
<div id="plotting-drug-dose-response-data" class="section level3">
<h3 class="hasAnchor">
<a href="#plotting-drug-dose-response-data" class="anchor"></a>Plotting Drug Dose Response Data</h3>
<p>Drug-Dose response data included in the PharmacoSet objects can be conviniently plotted using the <em>drugDoseResponseCurve</em> function. Given a list of PharmacoSets, a drug name and a cell name, it will plot the drug dose response curves for the given cell-drug combination in each dataset, allowing direct comparisons of data between datasets.</p>
<div class="sourceCode" id="cb15"><html><body><pre class="r"><span class="no">cells</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"SK-MEL-2"</span>,<span class="st">"697"</span>,<span class="st">"NCI-H1666"</span>)
<span class="fu"><a href="https://rdrr.io/r/graphics/par.html">par</a></span>(<span class="kw">mfrow</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>, <span class="fl">3</span>), <span class="kw">pty</span> <span class="kw">=</span> <span class="st">"s"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">drug</span><span class="kw">=</span><span class="st">"Lapatinib"</span>, <span class="kw">cellline</span><span class="kw">=</span><span class="no">cells</span>[<span class="fl">1</span>],
                      <span class="kw">pSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Actual"</span>,
                      <span class="kw">legends.label</span><span class="kw">=</span><span class="st">"aac_published"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">drug</span><span class="kw">=</span><span class="st">"Lapatinib"</span>, <span class="kw">cellline</span><span class="kw">=</span><span class="no">cells</span>[<span class="fl">2</span>],
                      <span class="kw">pSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Actual"</span>,
                      <span class="kw">legends.label</span><span class="kw">=</span><span class="st">"aac_published"</span>)
<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">drug</span><span class="kw">=</span><span class="st">"Lapatinib"</span>, <span class="kw">cellline</span><span class="kw">=</span><span class="no">cells</span>[<span class="fl">3</span>],
                      <span class="kw">pSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">CCLE</span>, <span class="no">GDSC</span>), <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Actual"</span>,
                      <span class="kw">legends.label</span><span class="kw">=</span><span class="st">"aac_published"</span>)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/curves-1.png" width="700"></p>
<p>The function <em>drugDoseResponseCurve</em> can also be used to plot your own drug dose response curves, as follows:</p>
<div class="sourceCode" id="cb16"><html><body><pre class="r"><span class="no">concentrations</span> <span class="kw">&lt;-</span> <span class="fl">1</span>/<span class="fl">2</span>^<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">8</span>) * <span class="fl">1</span>
<span class="no">viabilities</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">33.3</span>, <span class="fl">60</span>, <span class="fl">77.8</span>, <span class="fl">88.2</span>, <span class="fl">93.9</span>, <span class="fl">96.9</span>, <span class="fl">98.4</span>, <span class="fl">99.2</span>)

<span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugDoseResponseCurve.html">drugDoseResponseCurve</a></span>(<span class="kw">concentrations</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Exp 1"</span> <span class="kw">=</span> <span class="no">concentrations</span>), <span class="kw">viabilities</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Exp 1"</span> <span class="kw">=</span> <span class="no">viabilities</span>))</pre></body></html></div>
<pre><code>## Warning in sanitizeInput(concentrations[[i]], viabilities[[i]], conc_as_log =
## conc_as_log, : Concentration Values were unsorted. Sorting concentration and
## ordering viability in same order</code></pre>
<p><img src="GxWorkshop_files/figure-html/ddrc_own-1.png" width="700"></p>
<div id="fitting-drug-dose-response-curves" class="section level4">
<h4 class="hasAnchor">
<a href="#fitting-drug-dose-response-curves" class="anchor"></a>Fitting Drug Dose Response Curves</h4>
<p>One of the core tasks implemented in PharmacoGx is the fitting of Hill Curve Models to dose-response data. In PharmacoGx, we use the 3 Parameter Hill Curve function as our model of drug response in cancer cell lines:</p>
<p><span class="math display">\[y = E_\infty + \frac{1-E_\infty}{1 + (\frac{x}{EC50})^{HS}}\]</span> This is a log-logistic model, meaning that it takes the form of a logistic curve when x is on the log scale. The three parameters are: <span class="math inline">\(E_infty\)</span>, which is the maximal inhibition predicted at infinite concetration of the drug (the right asymptote); <span class="math inline">\(EC50\)</span>, which is the inflection point of the logit, where the inhibition reaches 50% of the maximum; <span class="math inline">\(HS\)</span>, the Hill Slope, which is a measure of the cooperativity of binding. This parameter controls the steepness of the logit, and is interpreted based on its relation to 1. Values larger than 1 imply positive cooperativity in inhibition/binding of the target for the small molcule ligands, which values less than 1 imply negative cooperativity/ antagonism.</p>
<p>To fit a Hill Slope model to your data in PharmacoGx, you can use the <em>logLogisticRegression</em> function, as below:</p>
<div class="sourceCode" id="cb18"><html><body><pre class="r"><span class="no">concentrations</span> <span class="kw">&lt;-</span> <span class="fl">1</span>/<span class="fl">2</span>^<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">8</span>) * <span class="fl">1</span>
<span class="no">viabilities</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">33.3</span>, <span class="fl">60</span>, <span class="fl">77.8</span>, <span class="fl">88.2</span>, <span class="fl">93.9</span>, <span class="fl">96.9</span>, <span class="fl">98.4</span>, <span class="fl">99.2</span>)

<span class="no">pars</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/logLogisticRegression.html">logLogisticRegression</a></span>(<span class="kw">conc</span> <span class="kw">=</span> <span class="no">concentrations</span>, <span class="kw">viability</span> <span class="kw">=</span> <span class="no">viabilities</span>)</pre></body></html></div>
<pre><code>## Warning in sanitizeInput(conc = conc, viability = viability, conc_as_log =
## conc_as_log, : Concentration Values were unsorted. Sorting concentration and
## ordering viability in same order</code></pre>
<div class="sourceCode" id="cb20"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">pars</span>)</pre></body></html></div>
<pre><code>## $HS
## [1] 1.900066
## 
## $E_inf
## [1] 0
## 
## $EC50
## [1] 0.2813364</code></pre>
</div>
</div>
<div id="computing-summary-measures-for-ddrcs" class="section level3">
<h3 class="hasAnchor">
<a href="#computing-summary-measures-for-ddrcs" class="anchor"></a>Computing Summary Measures for DDRCs</h3>
<p>Often with in vitro Pharmacogenomics data, we want to compare the drug sensitivity of a cell line to some omic feature. For this, we want to summarize the drug dose response curve into a single number representing the sensitivity of the cell line. The IC50 and Area Above the curve are two convenient metrics for quantifying the observed drug sensitivity. If you noticed above, the <em>drugDoseResponseCurve</em> function computes them by default. In PharmacoGx, they can be computed manually as follows:</p>
<div class="sourceCode" id="cb22"><html><body><pre class="r"><span class="no">concentrations</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span>(<span class="fl">1</span>/<span class="fl">2</span>^<span class="fu"><a href="https://rdrr.io/r/base/seq.html">seq</a></span>(<span class="fl">0</span>,<span class="fl">8</span>) * <span class="fl">1</span>)
<span class="no">viabilities</span> <span class="kw">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">99.2</span>, <span class="fl">98.4</span>, <span class="fl">96.9</span>, <span class="fl">93.9</span>, <span class="fl">88.2</span>, <span class="fl">77.8</span>, <span class="fl">60</span>, <span class="fl">33.3</span>, <span class="fl">0</span>)

<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeAUC.html">computeAUC</a></span>(<span class="kw">concentration</span> <span class="kw">=</span> <span class="no">concentrations</span>, <span class="kw">viability</span> <span class="kw">=</span> <span class="no">viabilities</span>))</pre></body></html></div>
<pre><code>## [1] 23.68416</code></pre>
<div class="sourceCode" id="cb24"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/computeICn.html">computeIC50</a></span>(<span class="kw">concentration</span> <span class="kw">=</span> <span class="no">concentrations</span>, <span class="kw">viability</span> <span class="kw">=</span> <span class="no">viabilities</span>))</pre></body></html></div>
<pre><code>## [1] 0.2813364</code></pre>
<p>In <em>PharmacoGx</em>, we call these measures sensitivity measures. PSets come with these measures precomputed, and accessible using the <em>sensitivityProfiles</em> function:</p>
<div class="sourceCode" id="cb26"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">sensitivityProfiles</span>(<span class="no">CCLE</span>))</pre></body></html></div>
<pre><code>##                        ic50_published aac_published amax_published
## drugid_AEW541_1321N1         8.000000     0.0873750      -42.55801
## drugid_AEW541_22Rv1          2.329924     0.2205000      -71.58934
## drugid_AEW541_42-MG-BA       2.682130     0.1144375      -63.49137
## drugid_AEW541_5637           5.002314     0.1243550      -62.35278
## drugid_AEW541_639-V          1.736181     0.1936250      -51.95981
## drugid_AEW541_697            4.260822     0.1087625      -73.33379
##                        aac_recomputed ic50_recomputed        HS    E_inf
## drugid_AEW541_1321N1       0.09800818       16.332263 0.9169196 38.47711
## drugid_AEW541_22Rv1        0.21323823        2.034710 0.8374151 13.95620
## drugid_AEW541_42-MG-BA     0.12151155        2.687672 2.9099240 36.28378
## drugid_AEW541_5637         0.10877695        5.192732 1.0883419  0.00000
## drugid_AEW541_639-V        0.18344576        5.113119 0.6991911 30.29307
## drugid_AEW541_697          0.10518547        4.254253 1.5039911  0.00000
##                            EC50
## drugid_AEW541_1321N1   3.295213
## drugid_AEW541_22Rv1    1.376470
## drugid_AEW541_42-MG-BA 1.723197
## drugid_AEW541_5637     5.192732
## drugid_AEW541_639-V    1.350123
## drugid_AEW541_697      4.254253</code></pre>
</div>
<div id="summary-functions" class="section level3">
<h3 class="hasAnchor">
<a href="#summary-functions" class="anchor"></a>Summary Functions</h3>
<p>Pharmacogenomics studies often contain many examples of either replicated or missing data. One of the most common tasks in preparing data for statistical or machine learning analysis is aligning your features and labels. To accelerate using Pharmacogenomics data for analysis, <em>PharmacoGx</em> contains two functions which create deduplicated matrices with missing data filled by NAs: <em>summarizeSensitivityProfiles</em> and <em>summarizeMolecularProfiles</em>. They create matrices which are drugs x cell lines and molecular features x cell lines, with any cell lines profiled for only sensitivity or molecular features padded with NA values. For the molecular profiles, the data is returned in a <code>r BiocStyle::Biocpkg("Biobase")</code> <em>ExpressionSet</em> object, while drug sensitivity data is returned as a matrix.</p>
<div class="sourceCode" id="cb28"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/summarizeMolecularProfiles.html">summarizeMolecularProfiles</a></span>(<span class="no">CCLE</span>, <span class="st">"rna"</span>))</pre></body></html></div>
<pre><code>## Summarizing rna molecular data for:  CCLE</code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |===================================                                   |  50%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code>## [1] 10012  1094</code></pre>
<div class="sourceCode" id="cb32"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/dim.html">dim</a></span>(<span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/summarizeSensitivityProfiles.html">summarizeSensitivityProfiles</a></span>(<span class="no">CCLE</span>, <span class="st">"aac_recomputed"</span>))</pre></body></html></div>
<pre><code>## [1]   24 1094</code></pre>
<p>Below, we use the summarizeSensitivityProfiles function and ggplot to investigate the distributions of AUC values within CCLE.</p>
<div class="sourceCode" id="cb34"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">ggplot2</span>, <span class="kw">verbose</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<pre><code>## Warning: package 'ggplot2' was built under R version 4.0.1</code></pre>
<div class="sourceCode" id="cb36"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">reshape2</span>, <span class="kw">verbose</span><span class="kw">=</span><span class="fl">FALSE</span>)
<span class="no">CCLE.aac</span> <span class="kw">&lt;-</span> <span class="kw pkg">PharmacoGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/summarizeSensitivityProfiles.html">summarizeSensitivityProfiles</a></span>(<span class="no">CCLE</span>, <span class="kw">sensitivity.measure</span> <span class="kw">=</span> <span class="st">"aac_recomputed"</span>)
<span class="no">melted_data</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/reshape2/man/melt.html">melt</a></span>(<span class="no">CCLE.aac</span>)
<span class="no">NA_rows</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html">unique</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/which.html">which</a></span>(<span class="fu"><a href="https://rdrr.io/r/base/NA.html">is.na</a></span>(<span class="no">melted_data</span>), <span class="kw">arr.ind</span><span class="kw">=</span><span class="no">T</span>)[,<span class="fl">1</span>])
<span class="no">melted_data</span> <span class="kw">&lt;-</span> <span class="no">melted_data</span>[-<span class="no">NA_rows</span>,]
<span class="no">p</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html">ggplot</a></span>(<span class="no">melted_data</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html">aes</a></span>(<span class="kw">x</span><span class="kw">=</span><span class="no">Var1</span>,<span class="kw">y</span><span class="kw">=</span><span class="no">value</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html">geom_boxplot</a></span>(<span class="kw">fill</span><span class="kw">=</span><span class="st">"gray"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html">theme</a></span>(<span class="kw">axis.text.x</span><span class="kw">=</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html">element_text</a></span>(<span class="kw">angle</span><span class="kw">=</span><span class="fl">90</span>,<span class="kw">hjust</span><span class="kw">=</span><span class="fl">1</span>)) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">xlab</a></span>(<span class="st">"Drugs"</span>) +
  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html">ylab</a></span>(<span class="st">"AAC"</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">p</span>)</pre></body></html></div>
<div class="figure">
<img src="GxWorkshop_files/figure-html/ccleaac-1.png" alt="Cells response to drugs in CCLE" width="700"><p class="caption">
Cells response to drugs in CCLE
</p>
</div>
</div>
<div id="signature-generation" class="section level3">
<h3 class="hasAnchor">
<a href="#signature-generation" class="anchor"></a>Signature Generation</h3>
<p>Finally, we will talk about some functions built into PharmacoGx to generate signatures of molecular features which correlate with response to particular compounds. In PharmacoGx, the main function to generate signatures of response across drugs is <code>drugSensitivitySig</code>. It takes as input a PSet, and the molecular data type you wish to use to generate the signatures, and computes the linear association between each feature and each drug response vector independently. The default of using all features, drugs and cell lines in the dataset can be modified with parameters detailed below.</p>
<p>At its core, drugSensitivitySig fits a nested linear model to the drug response data, using the molecular feature, as well as cell tissue of origin and batch effect as possible predictors. The function then uses ANOVA to judge whether including the molecular feature in the model leads to a statistically significant improvement in prediction accuracy (R^2). For more details on the implementation, we encourage readers to reference the PharmacoGx manuscript, or the vignettes included in the package.</p>
<p>Here, as an example, we will run <code>drugSensitivitySig</code> to generate a signature of response to Lapatinib and Erlotinib, two targetted therapies, from the GDSC data. We will do this with the first (largely random) 100 genes measured in the expression matrix, as the full computation can take some time to complete.</p>
<div class="sourceCode" id="cb37"><html><body><pre class="r"><span class="co">## Extracting the first 100 gene names. featureInfo brings up the rowData of the "rna" SummarizedExperiment</span>
<span class="no">chosen.genes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html">rownames</a></span>(<span class="fu">featureInfo</span>(<span class="no">GDSC</span>, <span class="st">"rna"</span>))[<span class="fl">100</span>]
<span class="no">sigs</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/PharmacoGx/man/drugSensitivitySig.html">drugSensitivitySig</a></span>(<span class="no">GDSC</span>, <span class="st">"rna"</span>, <span class="kw">sensitivity.measure</span> <span class="kw">=</span> <span class="st">"aac_recomputed"</span>, <span class="kw">drugs</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="st">"Erlotinib"</span>, <span class="st">"Lapatinib"</span>), <span class="kw">features</span> <span class="kw">=</span> <span class="no">chosen.genes</span>)</pre></body></html></div>
<pre><code>## Summarizing rna molecular data for:  GDSC_v1</code></pre>
<pre><code>## 
  |                                                                            
  |                                                                      |   0%
  |                                                                            
  |=                                                                     |   2%
  |                                                                            
  |==                                                                    |   3%
  |                                                                            
  |===                                                                   |   5%
  |                                                                            
  |=====                                                                 |   7%
  |                                                                            
  |======                                                                |   8%
  |                                                                            
  |=======                                                               |  10%
  |                                                                            
  |========                                                              |  11%
  |                                                                            
  |=========                                                             |  13%
  |                                                                            
  |==========                                                            |  15%
  |                                                                            
  |===========                                                           |  16%
  |                                                                            
  |=============                                                         |  18%
  |                                                                            
  |==============                                                        |  20%
  |                                                                            
  |===============                                                       |  21%
  |                                                                            
  |================                                                      |  23%
  |                                                                            
  |=================                                                     |  25%
  |                                                                            
  |==================                                                    |  26%
  |                                                                            
  |====================                                                  |  28%
  |                                                                            
  |=====================                                                 |  30%
  |                                                                            
  |======================                                                |  31%
  |                                                                            
  |=======================                                               |  33%
  |                                                                            
  |========================                                              |  34%
  |                                                                            
  |=========================                                             |  36%
  |                                                                            
  |==========================                                            |  38%
  |                                                                            
  |============================                                          |  39%
  |                                                                            
  |=============================                                         |  41%
  |                                                                            
  |==============================                                        |  43%
  |                                                                            
  |===============================                                       |  44%
  |                                                                            
  |================================                                      |  46%
  |                                                                            
  |=================================                                     |  48%
  |                                                                            
  |==================================                                    |  49%
  |                                                                            
  |====================================                                  |  51%
  |                                                                            
  |=====================================                                 |  52%
  |                                                                            
  |======================================                                |  54%
  |                                                                            
  |=======================================                               |  56%
  |                                                                            
  |========================================                              |  57%
  |                                                                            
  |=========================================                             |  59%
  |                                                                            
  |==========================================                            |  61%
  |                                                                            
  |============================================                          |  62%
  |                                                                            
  |=============================================                         |  64%
  |                                                                            
  |==============================================                        |  66%
  |                                                                            
  |===============================================                       |  67%
  |                                                                            
  |================================================                      |  69%
  |                                                                            
  |=================================================                     |  70%
  |                                                                            
  |==================================================                    |  72%
  |                                                                            
  |====================================================                  |  74%
  |                                                                            
  |=====================================================                 |  75%
  |                                                                            
  |======================================================                |  77%
  |                                                                            
  |=======================================================               |  79%
  |                                                                            
  |========================================================              |  80%
  |                                                                            
  |=========================================================             |  82%
  |                                                                            
  |===========================================================           |  84%
  |                                                                            
  |============================================================          |  85%
  |                                                                            
  |=============================================================         |  87%
  |                                                                            
  |==============================================================        |  89%
  |                                                                            
  |===============================================================       |  90%
  |                                                                            
  |================================================================      |  92%
  |                                                                            
  |=================================================================     |  93%
  |                                                                            
  |===================================================================   |  95%
  |                                                                            
  |====================================================================  |  97%
  |                                                                            
  |===================================================================== |  98%
  |                                                                            
  |======================================================================| 100%</code></pre>
<pre><code>## Computing drug sensitivity signatures...</code></pre>
</div>
</div>
</div>
<div id="basic-functionalities-of-radiogx" class="section level1">
<h1 class="hasAnchor">
<a href="#basic-functionalities-of-radiogx" class="anchor"></a>Basic Functionalities of RadioGx</h1>
<div id="installing-radiogx" class="section level2">
<h2 class="hasAnchor">
<a href="#installing-radiogx" class="anchor"></a>Installing RadioGx</h2>
<p>To install the <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> package, run:</p>
<div class="sourceCode" id="cb41"><html><body><pre class="r"><span class="kw pkg">devtools</span><span class="kw ns">::</span><span class="fu"><a href="https://devtools.r-lib.org//reference/remote-reexports.html">install_github</a></span>(<span class="st">'bhklab/RadioGx'</span>)
<span class="co"># BiocManager::install('RadioGx', version='devel') # Waiting on issue with Bioconductor credentials to be resolved so I can push updates</span></pre></body></html></div>
<div class="sourceCode" id="cb42"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/base/library.html">library</a></span>(<span class="no">RadioGx</span>)</pre></body></html></div>
</div>
<div id="radioset" class="section level2">
<h2 class="hasAnchor">
<a href="#radioset" class="anchor"></a>RadioSet</h2>
<p>The RadioSet has a structure similar to the <code>PharmacoSet</code> and also inherits from the CoreSet<a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> class . The radiation slot is implemented in <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> to hold relevant metadata about the type(s) of radiation used in the dose-response experiment, and is analogous to the drug slot in a <code>PharmacoSet</code>. The remainder of the slots mirror the <code>PharmacoSet</code>.</p>
<div class="figure">
<img src="RadioGxClassDiagram.png" alt="**RadioSet class diagram**. Objects comprising a `RadioSet` are enclosed in boxes. First box indicates type and name of each object. Second box indicates the structure of an object or class. Third box shows accessor methods from `RadioGx` for that specific object. '=&gt;' represents return and specifies what is returned from that item or method." width="1894"><p class="caption">
<strong>RadioSet class diagram</strong>. Objects comprising a <code>RadioSet</code> are enclosed in boxes. First box indicates type and name of each object. Second box indicates the structure of an object or class. Third box shows accessor methods from <code>RadioGx</code> for that specific object. ‘=&gt;’ represents return and specifies what is returned from that item or method.
</p>
</div>
</div>
<div id="downloading-an-rset" class="section level2">
<h2 class="hasAnchor">
<a href="#downloading-an-rset" class="anchor"></a>Downloading an RSet</h2>
<p><em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> provides an interface similar to <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em> and <em><a href="https://bioconductor.org/packages/3.12/Xeva">Xeva</a></em> for downloading our curated versions of published datasets.</p>
<p>To get a list of available <code>RadioSet</code>s, use:</p>
<div class="sourceCode" id="cb43"><html><body><pre class="r"><span class="no">RSets</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/availableRSets.html">availableRSets</a></span>()</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">RadioSet_name</th>
<th align="left">Date_updated</th>
<th align="left">DOI</th>
</tr></thead>
<tbody><tr class="odd">
<td align="left">Cleveland</td>
<td align="left">2020/06/16</td>
<td align="left">10.5281/zenodo.3899555</td>
</tr></tbody>
</table>
<p>As the <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> package was only recently released, there is currently only one dataset available. Let’s download the ‘Cleveland’ RSet, which contains a highly curated version of the data from Yang <em>et al.</em>, 2016.</p>
<div class="sourceCode" id="cb44"><html><body><pre class="r"><span class="no">Cleveland</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/downloadRSet.html">downloadRSet</a></span>(<span class="st">'Cleveland'</span>, <span class="kw">saveDir</span><span class="kw">=</span><span class="st">'.'</span>)</pre></body></html></div>
<div class="sourceCode" id="cb45"><html><body><pre class="r"><span class="no">Cleveland</span></pre></body></html></div>
<pre><code>## Name:  Cleveland 
## Date Created:  Wed Oct 25 17:38:42 2017 
## Number of cell lines:  540 
## Number of radiation types:  1 
## RNA: 
##  Dim:  20049 517 
## RNASeq: 
##  Dim:  61958 513 
## CNV: 
##  Dim:  24960 403 
## Drug pertubation: 
##  Please look at pertNumber(rSet) to determine number of
##             experiments for each radiation-cell combination.
## Drug sensitivity: 
##  Number of Experiments:  542 
##  Please look at sensNumber(rSet) to determine number of
##             experiments for each radiation-cell combination.</code></pre>
<p>Similar to PharmacoGx and Xeva, a summary of the contents of the RadioSet is printed when calling a RadioSet in the console. We can see that the Cleveland RSet contains sensitivity information for 540 cell-lines treated with a single type of radiation. The RSet also contains rna<a href="#fn2" class="footnote-ref" id="fnref2"><sup>2</sup></a>, rna-seq and cnv molecular data for a subset of available cell-lines.</p>
</div>
<div id="accessing-data" class="section level2">
<h2 class="hasAnchor">
<a href="#accessing-data" class="anchor"></a>Accessing Data</h2>
<p>RadioGx stores three major categories of data: metadata/annotations, molecular data and radiation response data. These are demarcated in <strong>Fig.</strong> @ref(fig:radioset) using green, blue and red, respectively. Accessor methods are available to retrieve all three kinds of data from an RSet; the accessor methods for each component are listed in the bottom most cell of each object in the RadioGx class diagram. We will discuss a subset of these methods now.</p>
<div id="accessing-metadata" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-metadata" class="anchor"></a>Accessing metadata</h3>
<p>Metadata in an RSet is stored in the same slots as in a <code>PharmacoSet</code>, and can be accessed using the same generic accessor functions as in <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em>. A unique slot, <code>radiation</code> has additional accessor methods to retrieve the radiation types used in a given sensitivity experiment.</p>
<div class="sourceCode" id="cb47"><html><body><pre class="r"><span class="co"># Get the radiation info data.frame from an RSet</span>
<span class="no">radInf</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/radiationInfo.html">radiationInfo</a></span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<div class="sourceCode" id="cb48"><html><body><pre class="r"><span class="kw pkg">knitr</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html">kable</a></span>(<span class="no">radInf</span>)</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="left">X.radiation.</th>
</tr></thead>
<tbody><tr class="odd">
<td>radiation</td>
<td align="left">radiation</td>
</tr></tbody>
</table>
<p>Currently, only one type of radiation has been used in an <code>RSet</code>. However, we hope to add new <code>RSet</code>s covering a wider range of radiation sensitivity and perturbation experiments in the near future. The following method is also available to retrieve the radiation types as a <code>character</code> vector instead of a <code>data.frame</code>.</p>
<div class="sourceCode" id="cb49"><html><body><pre class="r"><span class="no">radTypes</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/radiationTypes.html">radiationTypes</a></span>(<span class="no">Cleveland</span>)
<span class="no">radTypes</span></pre></body></html></div>
<pre><code>## [1] "radiation"</code></pre>
</div>
<div id="accessing-molecular-data" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-molecular-data" class="anchor"></a>Accessing molecular data</h3>
<p>Molecular data in an <code>RSet</code> is contained in the <code>molecularProfiles</code> slot and can be accessed the same way it is for a <code>PSet</code>.</p>
<div class="sourceCode" id="cb51"><html><body><pre class="r"><span class="co"># Get the list (equivalent to @molecularProfiles, except that it is robust to changes in RSet structure</span>
<span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="fu">molecularProfilesSlot</span>(<span class="no">Cleveland</span>), <span class="kw">max.level</span><span class="kw">=</span><span class="fl">2</span>)</pre></body></html></div>
<pre><code>## List of 4
##  $ rna     :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##  $ rnaseq  :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##  $ mutation:Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots
##  $ cnv     :Formal class 'SummarizedExperiment' [package "SummarizedExperiment"] with 5 slots</code></pre>
<div class="sourceCode" id="cb53"><html><body><pre class="r"><span class="co"># Get the names from the list</span>
<span class="fu">mDataNames</span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<pre><code>## [1] "rna"      "rnaseq"   "mutation" "cnv"</code></pre>
<p>All molecular data in an RSet (any class inheriting from CoreSet, actually) is contained in a <code>SummarizedExperiment</code> object. While <em><a href="https://bioconductor.org/packages/3.12/SummarizedExperiment">SummarizedExperiment</a></em> comes with it’s own set of accessors, we recommend using available <em><a href="https://bioconductor.org/packages/3.12/RadioGx">RadioGx</a></em> accessor methods as it allows your scripts to be robust to future changes in the structure of a <code>RadioSet</code> object.</p>
<div class="sourceCode" id="cb55"><html><body><pre class="r"><span class="co"># Get sample metadata</span>
<span class="no">phenoInf</span> <span class="kw">&lt;-</span> <span class="fu">phenoInfo</span>(<span class="no">Cleveland</span>, <span class="st">'cnv'</span>)</pre></body></html></div>
<p>To keep the document formatted nicely, the following tables have been subset to the first three rows and columns.</p>
<table class="table">
<thead><tr class="header">
<th align="left">Sample_title</th>
<th align="left">Sample_geo_accession</th>
<th align="left">Sample_status</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">A2780</td>
<td align="left">GSM887918</td>
<td align="left">Public on Mar 20 2012</td>
</tr>
<tr class="even">
<td align="left">OVCAR-4</td>
<td align="left">GSM888565</td>
<td align="left">Public on Mar 20 2012</td>
</tr>
<tr class="odd">
<td align="left">L3.3</td>
<td align="left">GSM888333</td>
<td align="left">Public on Mar 20 2012</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb56"><html><body><pre class="r"><span class="co"># Get feature metadata</span>
<span class="no">featInfo</span> <span class="kw">&lt;-</span> <span class="fu">featureInfo</span>(<span class="no">Cleveland</span>, <span class="st">'rna'</span>)</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th align="left">Probe</th>
<th align="left">EnsemblGeneId</th>
<th align="right">EntrezGeneId</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ENSG00000000003_at</td>
<td align="left">ENSG00000000003</td>
<td align="right">7105</td>
</tr>
<tr class="even">
<td align="left">ENSG00000000005_at</td>
<td align="left">ENSG00000000005</td>
<td align="right">64102</td>
</tr>
<tr class="odd">
<td align="left">ENSG00000000419_at</td>
<td align="left">ENSG00000000419</td>
<td align="right">8813</td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb57"><html><body><pre class="r"><span class="co"># Access the moleclar feature data</span>
<span class="no">mProf</span> <span class="kw">&lt;-</span> <span class="fu">molecularProfiles</span>(<span class="no">Cleveland</span>, <span class="st">'rnaseq'</span>)</pre></body></html></div>
<table class="table">
<thead><tr class="header">
<th></th>
<th align="right">G20461.HSC-3.2</th>
<th align="right">G20463.C2BBe1.2</th>
<th align="right">G20466.5637.2</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>ENSG00000000003</td>
<td align="right">4.527</td>
<td align="right">5.860</td>
<td align="right">5.394</td>
</tr>
<tr class="even">
<td>ENSG00000000005</td>
<td align="right">0.000</td>
<td align="right">0.070</td>
<td align="right">0.000</td>
</tr>
<tr class="odd">
<td>ENSG00000000419</td>
<td align="right">6.314</td>
<td align="right">7.303</td>
<td align="right">6.140</td>
</tr>
</tbody>
</table>
</div>
<div id="accessing-response-data" class="section level3">
<h3 class="hasAnchor">
<a href="#accessing-response-data" class="anchor"></a>Accessing response data</h3>
<p>Data from radiation sensitivity and/or perturbation experiments is also retrieved the same way it is for a PSet. Currently, only sensitivity experiments have been included in a <code>RadioSet</code>.</p>
<div class="sourceCode" id="cb58"><html><body><pre class="r"><span class="co"># Get sensitivity slot</span>
<span class="no">sens</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/sensitivitySlot.html">sensitivitySlot</a></span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<div class="sourceCode" id="cb59"><html><body><pre class="r"><span class="co"># Get sensitivity raw data</span>
<span class="no">sensRaw</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/sensitivityRaw.html">sensitivityRaw</a></span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<div class="sourceCode" id="cb60"><html><body><pre class="r"><span class="co"># Get sensitivity profiles</span>
<span class="no">sensProf</span> <span class="kw">&lt;-</span> <span class="fu">sensitivityProfiles</span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<p>``{r } # Get sensitivity info sensInfo &lt;- sensitivityInfo(Cleveland)</p>
<pre><code>
## Fitting Linear Quadratic (LQ) Models

RadioGx provides a number of functions for analyzing dose response experiments.
To use these functions, we must first fit a statistical model to the dose
response data. This package exports a function for fitting linear-quadratic
models to dose response data. The function can be used with data contained in
a RadioSet or with raw dose-response data.


```r
# Extract raw sensitvity data from the RadioSet
sensRaw &lt;- sensitivityRaw(Cleveland)</code></pre>
<div class="sourceCode" id="cb62"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/r/utils/str.html">str</a></span>(<span class="no">sensRaw</span>)</pre></body></html></div>
<pre><code>##  num [1:542, 1:9, 1:2] 1 1 1 1 1 1 1 1 1 1 ...
##  - attr(*, "dimnames")=List of 3
##   ..$ : chr [1:542] "SK-N-FI_radiation_1" "IMR-32_radiation_2" "SK-N-AS_radiation_3" "CHP-212_radiation_4" ...
##   ..$ : chr [1:9] "doses1" "doses2" "doses3" "doses4" ...
##   ..$ : chr [1:2] "Dose" "Viability"</code></pre>
<p>The data returned by <code><a href="https://rdrr.io/pkg/RadioGx/man/sensitivityRaw.html">sensitivityRaw(RSet)</a></code> is a three dimensional array, but it can also be thought of as a set of experiment by treatment matrices. We can see by the <code>dimnames</code> of the third dimensions that the first matrix holds the radiation dose (in Gy) for each experiment, while the second matrix holds the viability measurements for the cell-line after each dose in the experimental series.</p>
<div class="sourceCode" id="cb64"><html><body><pre class="r"><span class="co"># Find a cancer cell-line of interest</span>
<span class="fu"><a href="https://rdrr.io/r/utils/head.html">head</a></span>(<span class="fu">sensitivityInfo</span>(<span class="no">Cleveland</span>)$<span class="no">cellid</span>)</pre></body></html></div>
<pre><code>## [1] "SK-N-FI"   "IMR-32"    "SK-N-AS"   "CHP-212"   "KP-N-S19s" "MHH-NB-11"</code></pre>
<div class="sourceCode" id="cb66"><html><body><pre class="r"><span class="no">cancerCellLine</span> <span class="kw">&lt;-</span> <span class="fu">sensitivityInfo</span>(<span class="no">Cleveland</span>)$<span class="no">cellid</span>[<span class="fl">1</span>]
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">cancerCellLine</span>)</pre></body></html></div>
<pre><code>## [1] "SK-N-FI"</code></pre>
<div class="sourceCode" id="cb68"><html><body><pre class="r"><span class="co"># Get the radiation doses and associated survival data from clevelandSmall</span>
<span class="no">radiationDoses</span> <span class="kw">&lt;-</span> <span class="no">sensRaw</span>[<span class="fl">1</span>, , <span class="st">'Dose'</span>]
<span class="no">survivalFractions</span> <span class="kw">&lt;-</span> <span class="no">sensRaw</span>[<span class="fl">1</span>, , <span class="st">'Viability'</span>]</pre></body></html></div>
<div class="sourceCode" id="cb69"><html><body><pre class="r"><span class="no">LQmodel</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/linearQuadraticModel.html">linearQuadraticModel</a></span>(<span class="kw">D</span><span class="kw">=</span><span class="no">radiationDoses</span>,
                                <span class="kw">SF</span><span class="kw">=</span><span class="no">survivalFractions</span>)
<span class="no">LQmodel</span></pre></body></html></div>
<pre><code>##      alpha       beta 
## 0.15219719 0.01783452 
## attr(,"Rsquare")
## [1] 0.9882769</code></pre>
<p>Above we see that LQmodel contains the alpha and beta coefficients for the dose response curve fit to the dose and viability data for the IMR-32 cancer cell-line. Based on the <span class="math inline">\(R^2\)</span> attribute we can see that the model fit for this data is good, with 98.8% of observed variance explained by the model.</p>
</div>
</div>
<div id="calculating-dose-response-metrics" class="section level2">
<h2 class="hasAnchor">
<a href="#calculating-dose-response-metrics" class="anchor"></a>Calculating Dose-Response Metrics</h2>
<p>RadioGx provides a number of functions for calculating common dose response metrics such as surviving fraction (SF), area under the curve (AUC) and dose at which only 10% of cancer cells survive (D10).</p>
<p>Some of these functions require the alpha and beta coefficients, as calculated above using the <code>linearQuadraticModel</code> function.</p>
<div class="sourceCode" id="cb71"><html><body><pre class="r"><span class="no">survFracAfter2Units</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/computeSF2.html">computeSF2</a></span>(<span class="kw">pars</span><span class="kw">=</span><span class="no">LQmodel</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">survFracAfter2Units</span>)</pre></body></html></div>
<pre><code>## [1] 0.6867861</code></pre>
<div class="sourceCode" id="cb73"><html><body><pre class="r"><span class="no">dose10PercentSurv</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/computeD10.html">computeD10</a></span>(<span class="kw">pars</span><span class="kw">=</span><span class="no">LQmodel</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">dose10PercentSurv</span>)</pre></body></html></div>
<pre><code>## [1] 7.870413</code></pre>
<p>We see from the above code cell that after two units of radiation, 68.679% of cancer cells remain relative to the initial population. Conversely, using <code>computeD10</code> we see that on average 7.87 units of radiation need to be administered to result in 10% cell-line survival (i.e., 90% of cancer cells are killed).</p>
<p>Other dose-response metrics can be computed directly using radiation dose and cancer cell viability data.</p>
<div class="sourceCode" id="cb75"><html><body><pre class="r"><span class="no">areaUnderDoseRespCurve</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/computeAUC.html">computeAUC</a></span>(<span class="kw">D</span><span class="kw">=</span><span class="no">radiationDoses</span>, <span class="kw">pars</span><span class="kw">=</span><span class="no">LQmodel</span>, <span class="kw">lower</span><span class="kw">=</span><span class="fl">0</span>,
                                     <span class="kw">upper</span><span class="kw">=</span><span class="fl">1</span>)
<span class="fu"><a href="https://rdrr.io/r/base/print.html">print</a></span>(<span class="no">areaUnderDoseRespCurve</span>)</pre></body></html></div>
<pre><code>## [1] 2.872288</code></pre>
<p>In the above code block we compute the AUC for a dose-response curve between a dose of 0 to 1 Gy. This area can be interpreted as the total proportion of cells killed during the administration of 1 Gy of radiation.</p>
<div id="dose-response-curves" class="section level3">
<h3 class="hasAnchor">
<a href="#dose-response-curves" class="anchor"></a>Dose-Response Curves</h3>
<p>The <code>doseResponseCurve</code> function can be used to generate plots of surviving fraction vs dose for radiation sensitivity experiments. In this example we provide raw data values to create the plot. When the <code>plot.type</code> is set to “Both”, a linear-quadratic model will also be fit to the supplied dose-response values.</p>
<div class="sourceCode" id="cb77"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/doseResponseCurve.html">doseResponseCurve</a></span>(
  <span class="kw">Ds</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Experiment 1"</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">0</span>, <span class="fl">2</span>, <span class="fl">4</span>, <span class="fl">6</span>)),
  <span class="kw">SFs</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="st">"Experiment 1"</span> <span class="kw">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span>(<span class="fl">1</span>,<span class="fl">.6</span>,<span class="fl">.4</span>,<span class="fl">.2</span>)),
  <span class="kw">plot.type</span><span class="kw">=</span><span class="st">"Both"</span>
)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/plotting_rad_dose_resp-1.png" width="700"></p>
<p>Additionally, <code>doseResponseCurve</code> can be used to create dose response curves directly from a curated RadioSet object. When utilizing this feature, a cell-line must be selected from the RadioSet. This can be done by name if you know which cell-line you are looking for. If you don’t know which cell-line you want to visualize, the available cell-lines can be explored using the <code>cellInfo</code> function.</p>
<div class="sourceCode" id="cb78"><html><body><pre class="r"><span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/doseResponseCurve.html">doseResponseCurve</a></span>(
  <span class="kw">rSets</span><span class="kw">=</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span>(<span class="no">clevelandSmall</span>),
  <span class="kw">cellline</span><span class="kw">=</span><span class="fu">cellInfo</span>(<span class="no">clevelandSmall</span>)$<span class="no">cellid</span>[<span class="fl">3</span>]
)</pre></body></html></div>
<p><img src="GxWorkshop_files/figure-html/plotting_rad_dose_resp_rSet-1.png" width="700"></p>
</div>
<div id="summarizing-sensitivity" class="section level3">
<h3 class="hasAnchor">
<a href="#summarizing-sensitivity" class="anchor"></a>Summarizing Sensitivity</h3>
<p>To retrieve a radiation type by cell-line summary of a sensitivity experiment, we use the <code>summarizeSensitivityProfiles</code> function. This will return a <code>matrix</code> where rows are radiation type<a href="#fn3" class="footnote-ref" id="fnref3"><sup>3</sup></a>, columns are cell-line and values are viability measurements summarized using <code>summary.stat</code><a href="#fn4" class="footnote-ref" id="fnref4"><sup>4</sup></a>. The sensitivity measure to summarize can be specified using <code>sensitivity.measure</code><a href="#fn5" class="footnote-ref" id="fnref5"><sup>5</sup></a>.</p>
<div class="sourceCode" id="cb79"><html><body><pre class="r"><span class="no">sensSummary</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/summarizeSensitivityProfiles.html">summarizeSensitivityProfiles</a></span>(<span class="no">Cleveland</span>)</pre></body></html></div>
<div class="sourceCode" id="cb80"><html><body><pre class="r"><span class="no">sensSummary</span>[, <span class="fl">1</span>:<span class="fl">3</span>]</pre></body></html></div>
<pre><code>##   SK-N-FI    IMR-32   SK-N-AS 
## 2.8616423 0.5490508 3.2589305</code></pre>
</div>
<div id="summarizing-molecular-data" class="section level3">
<h3 class="hasAnchor">
<a href="#summarizing-molecular-data" class="anchor"></a>Summarizing Molecular Data</h3>
<div class="sourceCode" id="cb82"><html><body><pre class="r"><span class="no">mprofSummary</span> <span class="kw">&lt;-</span> <span class="kw pkg">RadioGx</span><span class="kw ns">::</span><span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/summarizeMolecularProfiles.html">summarizeMolecularProfiles</a></span>(<span class="no">Cleveland</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">'rna'</span>, <span class="kw">summary.stat</span><span class="kw">=</span><span class="st">'median'</span>, <span class="kw">fill.missing</span><span class="kw">=</span><span class="fl">FALSE</span>)</pre></body></html></div>
<div class="sourceCode" id="cb83"><html><body><pre class="r"><span class="no">mprofSummary</span></pre></body></html></div>
<pre><code>## class: SummarizedExperiment 
## dim: 20049 517 
## metadata(3): experimentData annotation protocolData
## assays(2): exprs se.exprs
## rownames(20049): ENSG00000000003 ENSG00000000005 ... ENSG00000280439
##   ENSG00000280448
## rowData names(7): Probe EnsemblGeneId ... BEST rownames
## colnames(517): SK-N-FI IMR-32 ... Ishikawa (Heraklio) 02 ER- T.T
## colData names(25): samplename filename ... rownames tissueid</code></pre>
<p>Due to a lack of replicates in the Cleveland <code>RSet</code>, the returned <code>SummarizedExperiment</code> object contains the same information as the original. For other experiments with replicates, however, the result should contain one column per unique cell-line id. For ease of interoperability with the response data contained in an <code>RSet</code>, if <code>fill.missing</code> is FALSE empty columns for the cell-lines in the sensitivity experiment, but not in the molecular profile will be added to the <code>SummarizedExperiment</code> such that the dimensions are equal.</p>
</div>
<div id="molecular-signatures-for-biomarker-discovery" class="section level3">
<h3 class="hasAnchor">
<a href="#molecular-signatures-for-biomarker-discovery" class="anchor"></a>Molecular Signatures for Biomarker Discovery</h3>
<p>The true usefulness of the <code>RadioGx</code> packages comes from the ability to determine gene signatures for a cell-lines from a sensitivity experiment. Cell-lines of interest to a given researcher can be selected and a molecular signature computed which correlates specific molecular features with a given sensitivity profile. Using this method one could identify signatures associated with either radio-sensitivity or radio-resistance. Combining this signature with drug response gene signatures from <em><a href="https://bioconductor.org/packages/3.12/PharmacoGx">PharmacoGx</a></em>, as will be done in the subsequent section, one can identify drugs which could augment the effectiveness of a given radiation signature. Perhaps more powerfully, drugs which target features associated with radio-resistance can be found, potentially synergistically increasing the overall effectiveness of the combine treatment.</p>
<div class="sourceCode" id="cb85"><html><body><pre class="r"><span class="no">radSensSig</span> <span class="kw">&lt;-</span> <span class="fu"><a href="https://rdrr.io/pkg/RadioGx/man/radSensitivitySig.html">radSensitivitySig</a></span>(<span class="no">Cleveland</span>, <span class="kw">mDataType</span><span class="kw">=</span><span class="st">'rna'</span>, <span class="kw">features</span><span class="kw">=</span><span class="fu">fNames</span>(<span class="no">Cleveland</span>, <span class="st">'rna'</span>)[<span class="fl">2</span>:<span class="fl">5</span>], <span class="kw">nthread</span><span class="kw">=</span><span class="fl">15</span>)</pre></body></html></div>
<pre><code>## Computing radiation sensitivity signatures...</code></pre>
<div class="sourceCode" id="cb87"><html><body><pre class="r"><span class="no">radSensSig</span>@<span class="kw">.Data</span></pre></body></html></div>
<pre><code>## , , estimate
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA
## 
## , , se
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA
## 
## , , n
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA
## 
## , , tstat
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA
## 
## , , fstat
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA
## 
## , , pvalue
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA
## 
## , , df
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA
## 
## , , fdr
## 
##                 radiation
## ENSG00000000005        NA
## ENSG00000000419        NA
## ENSG00000000457        NA
## ENSG00000000460        NA</code></pre>
</div>
</div>
</div>
<div id="data-analysis-walkthrough" class="section level1">
<h1 class="hasAnchor">
<a href="#data-analysis-walkthrough" class="anchor"></a>Data Analysis Walkthrough</h1>
</div>
<div id="understanding-analysis-results-and-synergies" class="section level1">
<h1 class="hasAnchor">
<a href="#understanding-analysis-results-and-synergies" class="anchor"></a>Understanding Analysis Results and Synergies</h1>
</div>
<div id="use-case-discussion-translational-science" class="section level1">
<h1 class="hasAnchor">
<a href="#use-case-discussion-translational-science" class="anchor"></a>Use Case Discussion: Translational Science</h1>
</div>
<div class="footnotes">
<hr>
<ol>
<li id="fn1"><p>exported by <em><a href="https://bioconductor.org/packages/3.12/CoreGx">CoreGx</a></em><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
<li id="fn2"><p>microarray<a href="#fnref2" class="footnote-back">↩︎</a></p></li>
<li id="fn3"><p>Note there is currently only one radiation type available, but in the future there will be more<a href="#fnref3" class="footnote-back">↩︎</a></p></li>
<li id="fn4"><p>Available options are ‘mean’, ‘median’, ‘first’, ‘last’<a href="#fnref4" class="footnote-back">↩︎</a></p></li>
<li id="fn5"><p>See <code>sensitivityMeasures(RSet)</code> for available options<a href="#fnref5" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Petr Smirnov.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.5.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
